<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Programmatically access the NIH / NCI Genomic Data Commons
        RESTful service.">
<title>NIH / NCI Genomic Data Commons Access • GenomicDataCommons</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="NIH / NCI Genomic Data Commons Access">
<meta property="og:description" content="Programmatically access the NIH / NCI Genomic Data Commons
        RESTful service.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">GenomicDataCommons</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.21.5</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/overview.html">The GenomicDataCommons Package</a>
    <a class="dropdown-item" href="articles/questions-and-answers.html">Questions and answers from over the years</a>
    <a class="dropdown-item" href="articles/somatic_mutations.html">Working with simple somatic mutations</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/Bioconductor/GenomicDataCommons/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="genomicdatacommons">GenomicDataCommons<a class="anchor" aria-label="anchor" href="#genomicdatacommons"></a>
</h1></div>
<!-- badges: start -->
</div>
<div class="section level1">
<h1 id="what-is-the-gdc">What is the GDC?<a class="anchor" aria-label="anchor" href="#what-is-the-gdc"></a>
</h1>
<p>From the <a href="https://gdc.nci.nih.gov/about-gdc" class="external-link">Genomic Data Commons (GDC) website</a>:</p>
<p>The National Cancer Institute’s (NCI’s) Genomic Data Commons (GDC) is a data sharing platform that promotes precision medicine in oncology. It is not just a database or a tool; it is an expandable knowledge network supporting the import and standardization of genomic and clinical data from cancer research programs.</p>
<p>The GDC contains NCI-generated data from some of the largest and most comprehensive cancer genomic datasets, including The Cancer Genome Atlas (TCGA) and Therapeutically Applicable Research to Generate Effective Therapies (TARGET). For the first time, these datasets have been harmonized using a common set of bioinformatics pipelines, so that the data can be directly compared.</p>
<p>As a growing knowledge system for cancer, the GDC also enables researchers to submit data, and harmonizes these data for import into the GDC. As more researchers add clinical and genomic data to the GDC, it will become an even more powerful tool for making discoveries about the molecular basis of cancer that may lead to better care for patients.</p>
<p>The <a href="https://gdc.cancer.gov/developers/gdc-data-model/gdc-data-model-components" class="external-link">data model for the GDC is complex</a>, but it worth a quick overview. The data model is encoded as a so-called property graph. Nodes represent entities such as Projects, Cases, Diagnoses, Files (various kinds), and Annotations. The relationships between these entities are maintained as edges. Both nodes and edges may have Properties that supply instance details. The GDC API exposes these nodes and edges in a somewhat simplified set of <a href="https://en.wikipedia.org/wiki/Representational_state_transfer" class="external-link">RESTful</a> endpoints.</p>
</div>
<div class="section level1">
<h1 id="quickstart">Quickstart<a class="anchor" aria-label="anchor" href="#quickstart"></a>
</h1>
<p>This software is available at Bioconductor.org and can be downloaded via <code><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">BiocManager::install</a></code>.</p>
<p>To report bugs or problems, either <a href="https://github.com/Bioconductor/GenomicDataCommons/issues" class="external-link">submit a new issue</a> or submit a <code>bug.report(package='GenomicDataCommons')</code> from within R (which will redirect you to the new issue on GitHub).</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Installation can be achieved via Bioconductor’s <code>BiocManager</code> package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">'GenomicDataCommons'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicDataCommons" class="external-link">GenomicDataCommons</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="check-basic-functionality">Check basic functionality<a class="anchor" aria-label="anchor" href="#check-basic-functionality"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/status.html">status</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; $commit</span></span>
<span><span class="co">#&gt; [1] "4dd3680528a19ed33cfc83c7d049426c97bb903b"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $data_release</span></span>
<span><span class="co">#&gt; [1] "Data Release 34.0 - July 27, 2022"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $status</span></span>
<span><span class="co">#&gt; [1] "OK"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $tag</span></span>
<span><span class="co">#&gt; [1] "3.0.0"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $version</span></span>
<span><span class="co">#&gt; [1] 1</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="find-data">Find data<a class="anchor" aria-label="anchor" href="#find-data"></a>
</h2>
<p>The following code builds a <code>manifest</code> that can be used to guide the download of raw data. Here, filtering finds gene expression files quantified as raw counts using <code>STAR</code> from ovarian cancer patients.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ge_manifest</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/query.html">files</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="reference/filtering.html">filter</a></span><span class="op">(</span> <span class="va">cases.project.project_id</span> <span class="op">==</span> <span class="st">'TCGA-OV'</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="reference/filtering.html">filter</a></span><span class="op">(</span> <span class="va">type</span> <span class="op">==</span> <span class="st">'gene_expression'</span> <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="reference/filtering.html">filter</a></span><span class="op">(</span> <span class="va">analysis.workflow_type</span> <span class="op">==</span> <span class="st">'STAR - Counts'</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="reference/manifest.html">manifest</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="va">ge_manifest</span></span>
<span><span class="co">#&gt;                                     id data_format     access                                                                   file_name</span></span>
<span><span class="co">#&gt; 1 7c69529f-2273-4dc4-b213-e84924d78bea         TSV       open d6472bd0-b4e2-4ed1-a892-e1702c195dc7.rna_seq.augmented_star_gene_counts.tsv</span></span>
<span><span class="co">#&gt; 2 0eff4634-f8c4-4db9-8a7c-331b21689bae         TSV       open 42165baf-b32c-4fc4-8b04-29c5b4e76de0.rna_seq.augmented_star_gene_counts.tsv</span></span>
<span><span class="co">#&gt; 3 7d74b4c5-6391-4b3e-95a3-020ea0869e86         TSV controlled   accf08d4-a784-4908-831a-7a08d4c5f0f5.rna_seq.star_splice_junctions.tsv.gz</span></span>
<span><span class="co">#&gt; 4 dc2aeea4-3cd0-4623-92f4-bbbc962851cc         TSV controlled   8ab508b9-2993-4e66-b8f9-81e32e936d4a.rna_seq.star_splice_junctions.tsv.gz</span></span>
<span><span class="co">#&gt; 5 0cf852be-d2e3-4fde-bba8-c93efae2961a         TSV       open 93831282-1dd1-49a3-acd7-dae2a49ca62e.rna_seq.augmented_star_gene_counts.tsv</span></span>
<span><span class="co">#&gt;                           submitter_id           data_category       acl            type file_size                 created_datetime                           md5sum</span></span>
<span><span class="co">#&gt; 1 7085a70b-2f63-4402-9e53-70f091f26fcb Transcriptome Profiling      open gene_expression   4254435 2021-12-13T20:53:42.329364-06:00 19d5596bba8949f4c138793608497d56</span></span>
<span><span class="co">#&gt; 2 f0d44930-b1ad-447a-86b9-27d0285954b9 Transcriptome Profiling      open gene_expression   4257461 2021-12-13T20:47:24.326497-06:00 d89d71b7c028c1643d7a3ee7857d8e01</span></span>
<span><span class="co">#&gt; 3 e6473134-6d65-414c-9f52-2c25057fac7d Transcriptome Profiling phs000178 gene_expression   3109435 2021-12-13T21:03:56.008440-06:00 fb8332d6413c44a9de02a1cbe6b018aa</span></span>
<span><span class="co">#&gt; 4 f99b93a9-70cb-44f8-bd1f-4edeee4425a4 Transcriptome Profiling phs000178 gene_expression   4607701 2021-12-13T21:02:23.944851-06:00 26231bed1ef67c093d3ce2b39def81cd</span></span>
<span><span class="co">#&gt; 5 fb4d7abe-b61a-4f35-9700-605f1bc1512f Transcriptome Profiling      open gene_expression   4265694 2021-12-13T20:50:55.234254-06:00 050763aabd36509f954137fbdc4eeb00</span></span>
<span><span class="co">#&gt;                   updated_datetime                              file_id                      data_type    state experimental_strategy</span></span>
<span><span class="co">#&gt; 1 2022-01-19T14:47:28.965154-06:00 7c69529f-2273-4dc4-b213-e84924d78bea Gene Expression Quantification released               RNA-Seq</span></span>
<span><span class="co">#&gt; 2 2022-01-19T14:47:07.478144-06:00 0eff4634-f8c4-4db9-8a7c-331b21689bae Gene Expression Quantification released               RNA-Seq</span></span>
<span><span class="co">#&gt; 3 2022-01-19T14:01:15.621847-06:00 7d74b4c5-6391-4b3e-95a3-020ea0869e86 Splice Junction Quantification released               RNA-Seq</span></span>
<span><span class="co">#&gt; 4 2022-01-19T14:01:15.621847-06:00 dc2aeea4-3cd0-4623-92f4-bbbc962851cc Splice Junction Quantification released               RNA-Seq</span></span>
<span><span class="co">#&gt; 5 2022-01-19T14:47:07.036781-06:00 0cf852be-d2e3-4fde-bba8-c93efae2961a Gene Expression Quantification released               RNA-Seq</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="download-data">Download data<a class="anchor" aria-label="anchor" href="#download-data"></a>
</h2>
<p>This code block downloads the 5 gene expression files specified in the query above. Using multiple processes to do the download very significantly speeds up the transfer in many cases. The following completes in about 15 seconds.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Bioconductor/BiocParallel" class="external-link">BiocParallel</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/register.html" class="external-link">register</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/MulticoreParam-class.html" class="external-link">MulticoreParam</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">destdir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">fnames</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">ge_manifest</span><span class="op">$</span><span class="va">id</span>,<span class="va">gdcdata</span><span class="op">)</span></span></code></pre></div>
<p>If the download had included controlled-access data, the download above would have needed to include a <code>token</code>. Details are available in <a href="#authentication">the authentication section below</a>.</p>
</div>
<div class="section level2">
<h2 id="metadata-queries">Metadata queries<a class="anchor" aria-label="anchor" href="#metadata-queries"></a>
</h2>
<p>Here we use a couple of ad-hoc helper functions to handle the output of the query. See the <code>inst/script/README.Rmd</code> folder for the source.</p>
<p>First, create a <code>data.frame</code> from the clinical data:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expands</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"diagnoses"</span>,<span class="st">"annotations"</span>,</span>
<span>             <span class="st">"demographic"</span>,<span class="st">"exposures"</span><span class="op">)</span></span>
<span><span class="va">clinResults</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/query.html">cases</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="reference/select.html">select</a></span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="reference/expand.html">expand</a></span><span class="op">(</span><span class="va">expands</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="reference/results.html">results</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">6</span><span class="op">)</span></span>
<span><span class="va">demoDF</span> <span class="op">&lt;-</span> <span class="fu">filterAllNA</span><span class="op">(</span><span class="va">clinResults</span><span class="op">$</span><span class="va">demographic</span><span class="op">)</span></span>
<span><span class="va">exposuresDF</span> <span class="op">&lt;-</span> <span class="fu">bindrowname</span><span class="op">(</span><span class="va">clinResults</span><span class="op">$</span><span class="va">exposures</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">demoDF</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span>
<span><span class="co">#&gt;                                      cause_of_death         race gender              ethnicity</span></span>
<span><span class="co">#&gt; 2525bfef-6962-4b7f-8e80-6186400ce624           &lt;NA&gt; not reported female           not reported</span></span>
<span><span class="co">#&gt; 126507c3-c0d7-41fb-9093-7deed5baf431 Cancer Related not reported female           not reported</span></span>
<span><span class="co">#&gt; c43ac461-9f03-44bc-be7d-3d867eb708a0           &lt;NA&gt; not reported female           not reported</span></span>
<span><span class="co">#&gt; a59a90d9-f1b0-49dd-9c97-bcaa6ba55d44 Cancer Related not reported   male           not reported</span></span>
<span><span class="co">#&gt; 59122a43-606a-4669-806b-6747e0ac9985           &lt;NA&gt;        white   male not hispanic or latino</span></span>
<span><span class="co">#&gt; 4447a969-e5c8-4291-b83c-53a0f7e77cbc Cancer Related        white female not hispanic or latino</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exposuresDF</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span>
<span><span class="co">#&gt;                                       submitter_id                 created_datetime    alcohol_intensity pack_years_smoked</span></span>
<span><span class="co">#&gt; 2525bfef-6962-4b7f-8e80-6186400ce624 C3N-03839-EXP 2019-12-30T10:23:07.190853-06:00 Lifelong Non-Drinker                NA</span></span>
<span><span class="co">#&gt; 126507c3-c0d7-41fb-9093-7deed5baf431 C3N-01518-EXP 2018-06-21T14:27:48.817254-05:00 Lifelong Non-Drinker                NA</span></span>
<span><span class="co">#&gt; c43ac461-9f03-44bc-be7d-3d867eb708a0 C3N-03933-EXP 2019-03-14T08:23:14.054975-05:00 Lifelong Non-Drinker                NA</span></span>
<span><span class="co">#&gt; a59a90d9-f1b0-49dd-9c97-bcaa6ba55d44 C3N-02695-EXP 2019-03-14T08:23:14.054975-05:00   Occasional Drinker              16.8</span></span>
<span><span class="co">#&gt; 59122a43-606a-4669-806b-6747e0ac9985 C3L-03642-EXP 2019-06-24T07:53:15.534197-05:00 Lifelong Non-Drinker              39.0</span></span>
<span><span class="co">#&gt; 4447a969-e5c8-4291-b83c-53a0f7e77cbc C3L-03728-EXP 2019-06-24T07:53:15.534197-05:00 Lifelong Non-Drinker                NA</span></span></code></pre></div>
<p>Note that the diagnoses data has multiple lines per patient:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">diagDF</span> <span class="op">&lt;-</span> <span class="fu">bindrowname</span><span class="op">(</span><span class="va">clinResults</span><span class="op">$</span><span class="va">diagnoses</span><span class="op">)</span></span>
<span><span class="va">diagDF</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span>
<span><span class="co">#&gt;                                      ajcc_pathologic_stage                 created_datetime tissue_or_organ_of_origin age_at_diagnosis</span></span>
<span><span class="co">#&gt; 2525bfef-6962-4b7f-8e80-6186400ce624             Stage IIB 2019-07-22T06:40:02.183501-05:00          Head of pancreas            19956</span></span>
<span><span class="co">#&gt; 126507c3-c0d7-41fb-9093-7deed5baf431          Not Reported 2018-12-03T12:05:16.846188-06:00             Temporal lobe            26312</span></span>
<span><span class="co">#&gt; c43ac461-9f03-44bc-be7d-3d867eb708a0             Stage III 2019-03-14T10:37:34.405260-05:00       Floor of mouth, NOS            25635</span></span>
<span><span class="co">#&gt; a59a90d9-f1b0-49dd-9c97-bcaa6ba55d44          Not Reported 2019-03-14T10:37:34.405260-05:00       Floor of mouth, NOS            16652</span></span>
<span><span class="co">#&gt; 59122a43-606a-4669-806b-6747e0ac9985          Not Reported 2019-07-22T06:40:02.183501-05:00          Upper lobe, lung            23384</span></span>
<span><span class="co">#&gt; 4447a969-e5c8-4291-b83c-53a0f7e77cbc          Not Reported 2019-05-07T07:41:33.411909-05:00              Frontal lobe            29326</span></span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="basic-design">Basic design<a class="anchor" aria-label="anchor" href="#basic-design"></a>
</h1>
<p>This package design is meant to have some similarities to the “tidyverse” approach of dplyr. Roughly, the functionality for finding and accessing files and metadata can be divided into:</p>
<ol style="list-style-type: decimal">
<li>Simple query constructors based on GDC API endpoints.</li>
<li>A set of verbs that when applied, adjust filtering, field selection, and faceting (fields for aggregation) and result in a new query object (an endomorphism)</li>
<li>A set of verbs that take a query and return results from the GDC</li>
</ol>
<p>In addition, there are auxiliary functions for asking the GDC API for information about available and default fields, slicing BAM files, and downloading actual data files. Here is an overview of functionality<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;See individual function and methods documentation for specific details.&lt;/p&gt;"><sup>1</sup></a>.</p>
<ul>
<li>Creating a query<ul>
<li><code><a href="reference/query.html">projects()</a></code></li>
<li><code><a href="reference/query.html">cases()</a></code></li>
<li><code><a href="reference/query.html">files()</a></code></li>
<li><code><a href="reference/query.html">annotations()</a></code></li>
</ul>
</li>
<li>Manipulating a query<ul>
<li><code><a href="reference/filtering.html">filter()</a></code></li>
<li><code><a href="reference/faceting.html">facet()</a></code></li>
<li><code><a href="reference/select.html">select()</a></code></li>
</ul>
</li>
<li>Introspection on the GDC API fields<ul>
<li><code><a href="reference/mapping.html">mapping()</a></code></li>
<li><code><a href="reference/available_fields.html">available_fields()</a></code></li>
<li><code><a href="reference/default_fields.html">default_fields()</a></code></li>
<li><code><a href="reference/grep_fields.html">grep_fields()</a></code></li>
<li><code><a href="reference/available_values.html">available_values()</a></code></li>
<li><code><a href="reference/available_expand.html">available_expand()</a></code></li>
</ul>
</li>
<li>Executing an API call to retrieve query results<ul>
<li><code><a href="reference/results.html">results()</a></code></li>
<li><code><a href="reference/count.html">count()</a></code></li>
<li><code><a href="reference/response.html">response()</a></code></li>
</ul>
</li>
<li>Raw data file downloads<ul>
<li><code><a href="reference/gdcdata.html">gdcdata()</a></code></li>
<li><code><a href="reference/transfer.html">transfer()</a></code></li>
<li><code><a href="reference/gdc_client.html">gdc_client()</a></code></li>
</ul>
</li>
<li>Summarizing and aggregating field values (faceting)<ul><li><code><a href="reference/aggregations.html">aggregations()</a></code></li></ul>
</li>
<li>Authentication<ul><li><code><a href="reference/gdc_token.html">gdc_token()</a></code></li></ul>
</li>
<li>BAM file slicing<ul><li><code><a href="reference/slicing.html">slicing()</a></code></li></ul>
</li>
</ul>
</div>


  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://www.bioconductor.org/packages/GenomicDataCommons" class="external-link">View on Bioconductor</a></li>
<li><a href="https://github.com/Bioconductor/GenomicDataCommons/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/Bioconductor/GenomicDataCommons/issues/new" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/Artistic-2.0" class="external-link">Artistic-2.0</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing GenomicDataCommons</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Martin Morgan <br><small class="roles"> Author </small>  </li>
<li>Sean Davis <br><small class="roles"> Author, maintainer </small>  </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/Bioconductor/GenomicDataCommons/actions" class="external-link"><img src="https://github.com/Bioconductor/GenomicDataCommons/workflows/R-CMD-check/badge.svg" alt="R-CMD-check"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Martin Morgan, Sean Davis.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
