<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Download one or more files from GDC. Files are downloaded using the
UUID and renamed to the file name on the remote system. By default,
neither the uuid nor the file name on the remote system can exist."><title>Download GDC files — gdcdata • GenomicDataCommons</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Download GDC files — gdcdata"><meta property="og:description" content="Download one or more files from GDC. Files are downloaded using the
UUID and renamed to the file name on the remote system. By default,
neither the uuid nor the file name on the remote system can exist."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">GenomicDataCommons</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.21.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/overview.html">The GenomicDataCommons Package</a>
    <a class="dropdown-item" href="../articles/questions-and-answers.html">Questions and answers from over the years</a>
    <a class="dropdown-item" href="../articles/somatic_mutations.html">Working with simple somatic mutations</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/Bioconductor/GenomicDataCommons/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Download GDC files</h1>
      <small class="dont-index">Source: <a href="https://github.com/Bioconductor/GenomicDataCommons/blob/HEAD/R/gdcdata.R" class="external-link"><code>R/gdcdata.R</code></a></small>
      <div class="d-none name"><code>gdcdata.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Download one or more files from GDC. Files are downloaded using the
UUID and renamed to the file name on the remote system. By default,
neither the uuid nor the file name on the remote system can exist.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">gdcdata</span><span class="op">(</span>
  <span class="va">uuids</span>,
  use_cached <span class="op">=</span> <span class="cn">TRUE</span>,
  progress <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span>,
  token <span class="op">=</span> <span class="cn">NULL</span>,
  access_method <span class="op">=</span> <span class="st">"api"</span>,
  transfer_args <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="va">...</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>uuids</dt>
<dd><p>character() of GDC file UUIDs.</p></dd>
<dt>use_cached</dt>
<dd><p>logical(1) default TRUE indicating that,
if found in the cache, the file will not be downloaded
again. If FALSE, all supplied uuids will be re-downloaded.</p></dd>
<dt>progress</dt>
<dd><p>logical(1) default TRUE in interactive sessions,
FALSE otherwise indicating whether a progress par should be
produced for each file download.</p></dd>
<dt>token</dt>
<dd><p>(optional) character(1) security token allowing access
to restricted data. See
<a href="https://gdc-docs.nci.nih.gov/API/Users_Guide/Authentication_and_Authorization/" class="external-link">https://gdc-docs.nci.nih.gov/API/Users_Guide/Authentication_and_Authorization/</a>.</p></dd>
<dt>access_method</dt>
<dd><p>character(1), either 'api' or 'client'. See details.</p></dd>
<dt>transfer_args</dt>
<dd><p>character(1), additional arguments to pass to
the gdc-client command line. See <code><a href="gdc_client.html">gdc_client</a></code> and
<code><a href="transfer.html">transfer_help</a></code> for details.</p></dd>
<dt>...</dt>
<dd><p>further arguments passed to files, particulary useful when
requesting <code>legacy=TRUE</code></p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>a named vector with file uuids as the names and paths as
the value</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function is appropriate for one or several files; for large
downloads use <code><a href="manifest.html">manifest</a></code> to create a manifest for and
the GDC Data Transfer Tool.</p>
<p>When access_method is "api", the GDC "data" endpoint is the
    transfer mechanism used. The alternative access_method, "client", will
    utilize the <code>gdc-client</code> transfer tool, which must be
    downloaded separately and available. See
    <code><a href="gdc_client.html">gdc_client</a></code> for details on specifying the location
    of the gdc-client executable.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="manifest.html">manifest</a></code> for downloading large data.</p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="co"># get some example file uuids</span></span>
<span class="r-in"><span class="va">uuids</span> <span class="op">&lt;-</span> <span class="fu"><a href="query.html">files</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">    <span class="fu"><a href="filtering.html">filter</a></span><span class="op">(</span><span class="op">~</span> <span class="va">access</span> <span class="op">==</span> <span class="st">'open'</span> <span class="op">&amp;</span> <span class="va">file_size</span> <span class="op">&lt;</span> <span class="fl">100000</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">    <span class="fu"><a href="results.html">results</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">    <span class="fu"><a href="ids.html">ids</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># and get the data, placing it into the gdc_cache() directory</span></span>
<span class="r-in"><span class="fu">gdcdata</span><span class="op">(</span><span class="va">uuids</span>, use_cached<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                              e4f53f7a-7bf4-4908-bbe7-4dee159e04b2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        "~/.cache/GenomicDataCommons/e4f53f7a-7bf4-4908-bbe7-4dee159e04b2/a50a7ba0-c479-410f-a338-9ab179975875.wxs.aliquot_ensemble_masked.maf.gz" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                              77f11c00-5e1c-4258-a5be-fd367459fadc </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "~/.cache/GenomicDataCommons/77f11c00-5e1c-4258-a5be-fd367459fadc/TCGA-BRCA.c6e7c59d-708f-497e-b6e6-ed91ba294943.ascat2.allelic_specific.seg.txt" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                              78026596-7c57-403d-85a3-ca244cbb9117 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                   "~/.cache/GenomicDataCommons/78026596-7c57-403d-85a3-ca244cbb9117/nationwidechildrens.org_ssf.TCGA-E9-A1NE.xml" </span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># legacy data</span></span>
<span class="r-in"><span class="va">exon</span> <span class="op">&lt;-</span> <span class="fu"><a href="query.html">files</a></span><span class="op">(</span>legacy <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">    <span class="fu"><a href="filtering.html">filter</a></span><span class="op">(</span> <span class="op">~</span> <span class="va">cases.project.project_id</span> <span class="op">==</span> <span class="st">"TCGA-COAD"</span> <span class="op">&amp;</span></span>
<span class="r-in">        <span class="va">data_category</span> <span class="op">==</span> <span class="st">"Gene expression"</span> <span class="op">&amp;</span></span>
<span class="r-in">        <span class="va">data_type</span> <span class="op">==</span> <span class="st">"Exon quantification"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span class="r-in">    <span class="fu"><a href="results.html">results</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="ids.html">ids</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="fu">gdcdata</span><span class="op">(</span><span class="va">exon</span>, legacy <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                               13f62ad4-b55d-4091-9cea-a4dce58f9943 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> "~/.cache/GenomicDataCommons/13f62ad4-b55d-4091-9cea-a4dce58f9943/unc.edu.85eb9996-655c-44f8-abbf-fbc332cf75cf.1756113.bt.exon_quantification.txt" </span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Martin Morgan, Sean Davis.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer></div>

  

  

  </body></html>

