<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This utility function allows quick text-based search of available
fields for using grep"><title>Find matching field names — grep_fields • GenomicDataCommons</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Find matching field names — grep_fields"><meta property="og:description" content="This utility function allows quick text-based search of available
fields for using grep"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">GenomicDataCommons</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.21.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/overview.html">The GenomicDataCommons Package</a>
    <a class="dropdown-item" href="../articles/questions-and-answers.html">Questions and answers from over the years</a>
    <a class="dropdown-item" href="../articles/somatic_mutations.html">Working with simple somatic mutations</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/Bioconductor/GenomicDataCommons/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Find matching field names</h1>
      <small class="dont-index">Source: <a href="https://github.com/Bioconductor/GenomicDataCommons/blob/HEAD/R/fields.R" class="external-link"><code>R/fields.R</code></a></small>
      <div class="d-none name"><code>grep_fields.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This utility function allows quick text-based search of available
fields for using <code><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></code></p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">grep_fields</span><span class="op">(</span><span class="va">entity</span>, <span class="va">pattern</span>, <span class="va">...</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>entity</dt>
<dd><p>one of the available gdc entities ('files','cases',...)
against which to gather available fields for matching</p></dd>


<dt>pattern</dt>
<dd><p>A regular expression that will be used
in a call to <code><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></code></p></dd>


<dt>...</dt>
<dd><p>passed on to grep</p></dd>


<dt>value</dt>
<dd><p>logical(1) whether to return values as opposed
to indices (passed along to grep)</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>character() vector of field names matching</p>
<p></p>
<p><code>pattern</code></p>


    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">grep_fields</span><span class="op">(</span><span class="st">'files'</span>,<span class="st">'analysis'</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1] "analysis.analysis_id"                                                     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2] "analysis.analysis_type"                                                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3] "analysis.created_datetime"                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4] "analysis.input_files.access"                                              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5] "analysis.input_files.average_base_quality"                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [6] "analysis.input_files.average_insert_size"                                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [7] "analysis.input_files.average_read_length"                                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [8] "analysis.input_files.channel"                                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [9] "analysis.input_files.chip_id"                                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [10] "analysis.input_files.chip_position"                                       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [11] "analysis.input_files.contamination"                                       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [12] "analysis.input_files.contamination_error"                                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [13] "analysis.input_files.created_datetime"                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [14] "analysis.input_files.data_category"                                       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [15] "analysis.input_files.data_format"                                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [16] "analysis.input_files.data_type"                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [17] "analysis.input_files.error_type"                                          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [18] "analysis.input_files.experimental_strategy"                               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [19] "analysis.input_files.file_id"                                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [20] "analysis.input_files.file_name"                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [21] "analysis.input_files.file_size"                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [22] "analysis.input_files.imaging_date"                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [23] "analysis.input_files.magnification"                                       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [24] "analysis.input_files.md5sum"                                              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [25] "analysis.input_files.mean_coverage"                                       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [26] "analysis.input_files.msi_score"                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [27] "analysis.input_files.msi_status"                                          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [28] "analysis.input_files.pairs_on_diff_chr"                                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [29] "analysis.input_files.plate_name"                                          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [30] "analysis.input_files.plate_well"                                          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [31] "analysis.input_files.platform"                                            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [32] "analysis.input_files.proc_internal"                                       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [33] "analysis.input_files.proportion_base_mismatch"                            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [34] "analysis.input_files.proportion_coverage_10X"                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [35] "analysis.input_files.proportion_coverage_10x"                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [36] "analysis.input_files.proportion_coverage_30X"                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [37] "analysis.input_files.proportion_coverage_30x"                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [38] "analysis.input_files.proportion_reads_duplicated"                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [39] "analysis.input_files.proportion_reads_mapped"                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [40] "analysis.input_files.proportion_targets_no_coverage"                      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [41] "analysis.input_files.read_pair_number"                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [42] "analysis.input_files.revision"                                            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [43] "analysis.input_files.stain_type"                                          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [44] "analysis.input_files.state"                                               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [45] "analysis.input_files.state_comment"                                       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [46] "analysis.input_files.submitter_id"                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [47] "analysis.input_files.total_reads"                                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [48] "analysis.input_files.tumor_ploidy"                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [49] "analysis.input_files.tumor_purity"                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [50] "analysis.input_files.updated_datetime"                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [51] "analysis.metadata.read_groups.RIN"                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [52] "analysis.metadata.read_groups.adapter_name"                               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [53] "analysis.metadata.read_groups.adapter_sequence"                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [54] "analysis.metadata.read_groups.base_caller_name"                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [55] "analysis.metadata.read_groups.base_caller_version"                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [56] "analysis.metadata.read_groups.chipseq_antibody"                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [57] "analysis.metadata.read_groups.chipseq_target"                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [58] "analysis.metadata.read_groups.created_datetime"                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [59] "analysis.metadata.read_groups.days_to_sequencing"                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [60] "analysis.metadata.read_groups.experiment_name"                            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [61] "analysis.metadata.read_groups.flow_cell_barcode"                          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [62] "analysis.metadata.read_groups.fragment_maximum_length"                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [63] "analysis.metadata.read_groups.fragment_mean_length"                       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [64] "analysis.metadata.read_groups.fragment_minimum_length"                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [65] "analysis.metadata.read_groups.fragment_standard_deviation_length"         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [66] "analysis.metadata.read_groups.fragmentation_enzyme"                       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [67] "analysis.metadata.read_groups.includes_spike_ins"                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [68] "analysis.metadata.read_groups.instrument_model"                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [69] "analysis.metadata.read_groups.is_paired_end"                              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [70] "analysis.metadata.read_groups.lane_number"                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [71] "analysis.metadata.read_groups.library_name"                               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [72] "analysis.metadata.read_groups.library_preparation_kit_catalog_number"     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [73] "analysis.metadata.read_groups.library_preparation_kit_name"               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [74] "analysis.metadata.read_groups.library_preparation_kit_vendor"             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [75] "analysis.metadata.read_groups.library_preparation_kit_version"            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [76] "analysis.metadata.read_groups.library_selection"                          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [77] "analysis.metadata.read_groups.library_strand"                             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [78] "analysis.metadata.read_groups.library_strategy"                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [79] "analysis.metadata.read_groups.multiplex_barcode"                          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [80] "analysis.metadata.read_groups.number_expect_cells"                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [81] "analysis.metadata.read_groups.platform"                                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [82] "analysis.metadata.read_groups.read_group_id"                              </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [83] "analysis.metadata.read_groups.read_group_name"                            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [84] "analysis.metadata.read_groups.read_group_qcs.adapter_content"             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [85] "analysis.metadata.read_groups.read_group_qcs.basic_statistics"            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [86] "analysis.metadata.read_groups.read_group_qcs.created_datetime"            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [87] "analysis.metadata.read_groups.read_group_qcs.encoding"                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [88] "analysis.metadata.read_groups.read_group_qcs.fastq_name"                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [89] "analysis.metadata.read_groups.read_group_qcs.kmer_content"                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [90] "analysis.metadata.read_groups.read_group_qcs.overrepresented_sequences"   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [91] "analysis.metadata.read_groups.read_group_qcs.per_base_n_content"          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [92] "analysis.metadata.read_groups.read_group_qcs.per_base_sequence_content"   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [93] "analysis.metadata.read_groups.read_group_qcs.per_base_sequence_quality"   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [94] "analysis.metadata.read_groups.read_group_qcs.per_sequence_gc_content"     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [95] "analysis.metadata.read_groups.read_group_qcs.per_sequence_quality_score"  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [96] "analysis.metadata.read_groups.read_group_qcs.per_tile_sequence_quality"   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [97] "analysis.metadata.read_groups.read_group_qcs.percent_gc_content"          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [98] "analysis.metadata.read_groups.read_group_qcs.read_group_qc_id"            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [99] "analysis.metadata.read_groups.read_group_qcs.sequence_duplication_levels" </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [100] "analysis.metadata.read_groups.read_group_qcs.sequence_length_distribution"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [101] "analysis.metadata.read_groups.read_group_qcs.state"                       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [102] "analysis.metadata.read_groups.read_group_qcs.submitter_id"                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [103] "analysis.metadata.read_groups.read_group_qcs.total_sequences"             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [104] "analysis.metadata.read_groups.read_group_qcs.updated_datetime"            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [105] "analysis.metadata.read_groups.read_group_qcs.workflow_end_datetime"       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [106] "analysis.metadata.read_groups.read_group_qcs.workflow_link"               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [107] "analysis.metadata.read_groups.read_group_qcs.workflow_start_datetime"     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [108] "analysis.metadata.read_groups.read_group_qcs.workflow_type"               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [109] "analysis.metadata.read_groups.read_group_qcs.workflow_version"            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [110] "analysis.metadata.read_groups.read_length"                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [111] "analysis.metadata.read_groups.rin"                                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [112] "analysis.metadata.read_groups.sequencing_center"                          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [113] "analysis.metadata.read_groups.sequencing_date"                            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [114] "analysis.metadata.read_groups.single_cell_library"                        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [115] "analysis.metadata.read_groups.size_selection_range"                       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [116] "analysis.metadata.read_groups.spike_ins_concentration"                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [117] "analysis.metadata.read_groups.spike_ins_fasta"                            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [118] "analysis.metadata.read_groups.state"                                      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [119] "analysis.metadata.read_groups.submitter_id"                               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [120] "analysis.metadata.read_groups.target_capture_kit"                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [121] "analysis.metadata.read_groups.target_capture_kit_catalog_number"          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [122] "analysis.metadata.read_groups.target_capture_kit_name"                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [123] "analysis.metadata.read_groups.target_capture_kit_target_region"           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [124] "analysis.metadata.read_groups.target_capture_kit_vendor"                  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [125] "analysis.metadata.read_groups.target_capture_kit_version"                 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [126] "analysis.metadata.read_groups.to_trim_adapter_sequence"                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [127] "analysis.metadata.read_groups.updated_datetime"                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [128] "analysis.state"                                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [129] "analysis.submitter_id"                                                    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [130] "analysis.updated_datetime"                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [131] "analysis.workflow_end_datetime"                                           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [132] "analysis.workflow_link"                                                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [133] "analysis.workflow_start_datetime"                                         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [134] "analysis.workflow_type"                                                   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [135] "analysis.workflow_version"                                                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [136] "cases.follow_ups.molecular_tests.molecular_analysis_method"               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [137] "downstream_analyses.analysis_id"                                          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [138] "downstream_analyses.analysis_type"                                        </span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Martin Morgan, Sean Davis.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

