<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content='The manifest function/method creates a manifest of files to be downloaded
using the GDC Data Transfer Tool. There are methods for
creating manifest data frames from GDCQuery objects
that contain file information ("cases" and "files" queries).'><title>Prepare GDC manifest file for bulk download — manifest • GenomicDataCommons</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Prepare GDC manifest file for bulk download — manifest"><meta property="og:description" content='The manifest function/method creates a manifest of files to be downloaded
using the GDC Data Transfer Tool. There are methods for
creating manifest data frames from GDCQuery objects
that contain file information ("cases" and "files" queries).'><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">GenomicDataCommons</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.23.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/overview.html">The GenomicDataCommons Package</a>
    <a class="dropdown-item" href="../articles/questions-and-answers.html">Questions and answers from over the years</a>
    <a class="dropdown-item" href="../articles/somatic_mutations.html">Working with simple somatic mutations</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/Bioconductor/GenomicDataCommons/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Prepare GDC manifest file for bulk download</h1>
      <small class="dont-index">Source: <a href="https://github.com/Bioconductor/GenomicDataCommons/blob/HEAD/R/manifest.R" class="external-link"><code>R/manifest.R</code></a></small>
      <div class="d-none name"><code>manifest.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>The <code>manifest</code> function/method creates a manifest of files to be downloaded
using the GDC Data Transfer Tool. There are methods for
creating manifest data frames from <code><a href="query.html">GDCQuery</a></code> objects
that contain file information ("cases" and "files" queries).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">manifest</span><span class="op">(</span><span class="va">x</span>, from <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fu"><a href="count.html">count</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for gdc_files</span></span>
<span><span class="fu">manifest</span><span class="op">(</span><span class="va">x</span>, from <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fu"><a href="count.html">count</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for GDCfilesResponse</span></span>
<span><span class="fu">manifest</span><span class="op">(</span><span class="va">x</span>, from <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fu"><a href="count.html">count</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for GDCcasesResponse</span></span>
<span><span class="fu">manifest</span><span class="op">(</span><span class="va">x</span>, from <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fu"><a href="count.html">count</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>x</dt>
<dd><p>An <code><a href="query.html">GDCQuery</a></code> object of subclass "gdc_files" or "gdc_cases".</p></dd>


<dt>from</dt>
<dd><p>Record number from which to start when returning the manifest.</p></dd>


<dt>size</dt>
<dd><p>The total number of records to return.  Default 
will return the usually desirable full set of records.</p></dd>


<dt>...</dt>
<dd><p>passed to <code><a href="https://httr.r-lib.org/reference/PUT.html" class="external-link">PUT</a></code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A <code><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></code>, also of type "gdc_manifest", with five columns:</p><ul><li><p>id</p></li>
<li><p>filename</p></li>
<li><p>md5</p></li>
<li><p>size</p></li>
<li><p>state</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="methods-by-class-">Methods (by class)<a class="anchor" aria-label="anchor" href="#methods-by-class-"></a></h2>
    
<ul><li><p><code>manifest(gdc_files)</code>:</p></li>
<li><p><code>manifest(GDCfilesResponse)</code>:</p></li>
<li><p><code>manifest(GDCcasesResponse)</code>:</p></li>
</ul></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">gFiles</span> <span class="op">=</span> <span class="fu"><a href="query.html">files</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">shortManifest</span> <span class="op">=</span> <span class="va">gFiles</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">manifest</span><span class="op">(</span>size<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">shortManifest</span>,n<span class="op">=</span><span class="fl">3</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                     id data_format     access</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 c786e2e9-04ea-48cf-b8a5-3daa96c1837b         BAM controlled</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 4f27d16f-0029-43f5-8627-7f477b4cce6d         TSV controlled</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 c7546d88-1748-4760-b06f-0b510484345a         VCF controlled</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                      file_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1     c7522bbe-48b8-4338-b36e-564021353f00.rna_seq.transcriptome.gdc_realn.bam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2    c7522bbe-48b8-4338-b36e-564021353f00.rna_seq.star_splice_junctions.tsv.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 bce47c3d-a414-49c9-bd27-9e941144860c.wgs.CaVEMan.raw_somatic_mutation.vcf.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           submitter_id               data_category       acl</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 83385997-2e30-4a0e-ba08-8a359cf9e758            Sequencing Reads phs001486</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 b1a12732-5668-4c9a-90b2-ed8817ffa063     Transcriptome Profiling phs001486</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 4d86c0a2-deb6-4060-9a64-1ad96ef94a51 Simple Nucleotide Variation phs001486</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      type platform   file_size                 created_datetime</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1           aligned_reads Illumina 16264880178 2022-01-05T13:58:13.190285-06:00</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2         gene_expression     &lt;NA&gt;     2744657 2022-01-05T16:06:14.065970-06:00</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 simple_somatic_mutation     &lt;NA&gt;     3126634 2020-07-10T11:39:51.205109-05:00</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             md5sum                 updated_datetime</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 c1eba4660cf44f9018fe9323bb4c88ec 2022-01-19T23:25:37.395824-06:00</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 7d52913e765f757485357bb7c7bedd19 2022-01-19T23:22:31.836567-06:00</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 23fa4cbce43707f7da9100815d699262 2020-10-26T12:35:23.058290-05:00</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                file_id                      data_type    state</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 c786e2e9-04ea-48cf-b8a5-3daa96c1837b                  Aligned Reads released</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 4f27d16f-0029-43f5-8627-7f477b4cce6d Splice Junction Quantification released</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 c7546d88-1748-4760-b06f-0b510484345a    Raw Simple Somatic Mutation released</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   experimental_strategy state_comment error_type proportion_reads_mapped</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1               RNA-Seq            NA         NA                      NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2               RNA-Seq            NA         NA                      NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3                   WGS            NA         NA                      NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   proportion_base_mismatch contamination_error pairs_on_diff_chr total_reads</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                       NA                  NA                NA          NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                       NA                  NA                NA          NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3                       NA                  NA                NA          NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   mean_coverage proportion_reads_duplicated contamination average_base_quality</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1            NA                          NA            NA                   NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2            NA                          NA            NA                   NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3            NA                          NA            NA                   NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   average_insert_size average_read_length proportion_targets_no_coverage</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                  NA                  NA                             NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                  NA                  NA                             NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3                  NA                  NA                             NA</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Martin Morgan, Sean Davis.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

