<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content='The manifest function/method creates a manifest of files to be downloaded
using the GDC Data Transfer Tool. There are methods for
creating manifest data frames from GDCQuery objects
that contain file information ("cases" and "files" queries).'><title>Prepare GDC manifest file for bulk download — manifest • GenomicDataCommons</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Prepare GDC manifest file for bulk download — manifest"><meta property="og:description" content='The manifest function/method creates a manifest of files to be downloaded
using the GDC Data Transfer Tool. There are methods for
creating manifest data frames from GDCQuery objects
that contain file information ("cases" and "files" queries).'><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">GenomicDataCommons</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.21.4</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/overview.html">The GenomicDataCommons Package</a>
    <a class="dropdown-item" href="../articles/questions-and-answers.html">Questions and answers from over the years</a>
    <a class="dropdown-item" href="../articles/somatic_mutations.html">Working with simple somatic mutations</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/Bioconductor/GenomicDataCommons/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Prepare GDC manifest file for bulk download</h1>
      <small class="dont-index">Source: <a href="https://github.com/Bioconductor/GenomicDataCommons/blob/HEAD/R/manifest.R" class="external-link"><code>R/manifest.R</code></a></small>
      <div class="d-none name"><code>manifest.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>The <code>manifest</code> function/method creates a manifest of files to be downloaded
using the GDC Data Transfer Tool. There are methods for
creating manifest data frames from <code><a href="query.html">GDCQuery</a></code> objects
that contain file information ("cases" and "files" queries).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">manifest</span><span class="op">(</span><span class="va">x</span>, from <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fu"><a href="count.html">count</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for gdc_files</span></span>
<span><span class="fu">manifest</span><span class="op">(</span><span class="va">x</span>, from <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fu"><a href="count.html">count</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for GDCfilesResponse</span></span>
<span><span class="fu">manifest</span><span class="op">(</span><span class="va">x</span>, from <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fu"><a href="count.html">count</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S3 method for GDCcasesResponse</span></span>
<span><span class="fu">manifest</span><span class="op">(</span><span class="va">x</span>, from <span class="op">=</span> <span class="fl">0</span>, size <span class="op">=</span> <span class="fu"><a href="count.html">count</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>x</dt>
<dd><p>An <code><a href="query.html">GDCQuery</a></code> object of subclass "gdc_files" or "gdc_cases".</p></dd>


<dt>from</dt>
<dd><p>Record number from which to start when returning the manifest.</p></dd>


<dt>size</dt>
<dd><p>The total number of records to return.  Default 
will return the usually desirable full set of records.</p></dd>


<dt>...</dt>
<dd><p>passed to <code><a href="https://httr.r-lib.org/reference/PUT.html" class="external-link">PUT</a></code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A <code><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></code>, also of type "gdc_manifest", with five columns:</p><ul><li><p>id</p></li>
<li><p>filename</p></li>
<li><p>md5</p></li>
<li><p>size</p></li>
<li><p>state</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="methods-by-class-">Methods (by class)<a class="anchor" aria-label="anchor" href="#methods-by-class-"></a></h2>
    
<ul><li><p><code>manifest(gdc_files)</code>:</p></li>
<li><p><code>manifest(GDCfilesResponse)</code>:</p></li>
<li><p><code>manifest(GDCcasesResponse)</code>:</p></li>
</ul></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">gFiles</span> <span class="op">=</span> <span class="fu"><a href="query.html">files</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">shortManifest</span> <span class="op">=</span> <span class="va">gFiles</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">manifest</span><span class="op">(</span>size<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">shortManifest</span>,n<span class="op">=</span><span class="fl">3</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                     id data_format     access</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 1294fc89-fae1-4d71-b970-dee321eeb6c1         VCF controlled</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 e1e7a77e-9d63-4247-b8f3-e2330215e85c         VCF controlled</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 6c59c1f9-8f35-41db-958b-a06047b24d01         VCF controlled</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                        file_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                      9160d8a8-006d-4fd5-8c9e-09ec6179369a.wxs.muse.raw_somatic_mutation.vcf.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                    e32336ff-87c2-4d5d-9366-ac835157d08d.wxs.pindel.raw_somatic_mutation.vcf.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 MMRF_COMMPASS.124d4920-7fe1-491d-93da-ad76f9eb8f75.wxs.SomaticSniper.somatic_annotation.vcf.gz</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           submitter_id               data_category       acl</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 8c7fcd35-5fda-4eff-b368-f373fefc8327 Simple Nucleotide Variation phs000748</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 40ac1eea-f152-403e-9a58-66dd15384d14 Simple Nucleotide Variation phs000748</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 43f6c8d5-835d-47ec-a7c8-22ac6a88c73b Simple Nucleotide Variation phs000748</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         type file_size state_comment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1    simple_somatic_mutation     29042            NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2    simple_somatic_mutation     47937            NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 annotated_somatic_mutation   2175870            NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   created_datetime                           md5sum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 2019-01-28T10:21:43.275213-06:00 0138a8506fdbca4cf5b6c9e71c61f0b7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 2019-07-15T10:23:04.984044-05:00 6a94c60d2483df6264f876bee28aa611</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 2022-01-26T21:35:47.361393-06:00 3fff8d28dfde7fb7140b1c25e84b8f6e</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   updated_datetime                              file_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 2019-06-24T08:07:19.797044-05:00 1294fc89-fae1-4d71-b970-dee321eeb6c1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 2019-08-21T12:47:37.999949-05:00 e1e7a77e-9d63-4247-b8f3-e2330215e85c</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 2022-01-30T17:20:50.104776-06:00 6c59c1f9-8f35-41db-958b-a06047b24d01</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   error_type                   data_type    state experimental_strategy</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1         NA Raw Simple Somatic Mutation released                   WXS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2         NA Raw Simple Somatic Mutation released                   WXS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3         NA  Annotated Somatic Mutation released                   WXS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   proportion_reads_mapped proportion_base_mismatch contamination_error platform</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                      NA                       NA                  NA     &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                      NA                       NA                  NA     &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3                      NA                       NA                  NA     &lt;NA&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   pairs_on_diff_chr total_reads mean_coverage proportion_reads_duplicated</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                NA          NA            NA                          NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                NA          NA            NA                          NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3                NA          NA            NA                          NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   contamination average_base_quality average_insert_size average_read_length</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1            NA                   NA                  NA                  NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2            NA                   NA                  NA                  NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3            NA                   NA                  NA                  NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   proportion_targets_no_coverage</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1                             NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2                             NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3                             NA</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Martin Morgan, Sean Davis.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

