<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="GenomicDataCommons">
<title>The GenomicDataCommons Package • GenomicDataCommons</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="The GenomicDataCommons Package">
<meta property="og:description" content="GenomicDataCommons">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">GenomicDataCommons</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.21.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/overview.html">The GenomicDataCommons Package</a>
    <a class="dropdown-item" href="../articles/questions-and-answers.html">Questions and answers from over the years</a>
    <a class="dropdown-item" href="../articles/somatic_mutations.html">Working with simple somatic mutations</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/Bioconductor/GenomicDataCommons/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="overview_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>The GenomicDataCommons Package</h1>
                        <h4 data-toc-skip class="author">Sean Davis &amp; Martin Morgan</h4>
            
            <h4 data-toc-skip class="date">Tuesday, August 02, 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/Bioconductor/GenomicDataCommons/blob/HEAD/vignettes/overview.Rmd" class="external-link"><code>vignettes/overview.Rmd</code></a></small>
      <div class="d-none name"><code>overview.Rmd</code></div>
    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>The National Cancer Institute (NCI) has established the <a href="https://gdc.nci.nih.gov/" class="external-link">Genomic Data Commons</a> (GDC). The GDC provides the cancer research community with an open and unified repository for sharing and accessing data across numerous cancer studies and projects via a high-performance data transfer and query infrastructure. The <em>GenomicDataCommons</em> Bioconductor package provides basic infrastructure for querying, accessing, and mining genomic datasets available from the GDC. We expect that the Bioconductor developer and the larger bioinformatics communities will build on the <em>GenomicDataCommons</em> package to add higher-level functionality and expose cancer genomics data to the plethora of state-of-the-art bioinformatics methods available in Bioconductor.</p>
    </div>
    
<div class="section level2">
<h2 id="what-is-the-gdc">What is the GDC?<a class="anchor" aria-label="anchor" href="#what-is-the-gdc"></a>
</h2>
<p>From the <a href="https://gdc.cancer.gov/about-gdc" class="external-link">Genomic Data Commons (GDC) website</a>:</p>
<blockquote>
<p>The National Cancer Institute’s (NCI’s) Genomic Data Commons (GDC) is a data sharing platform that promotes precision medicine in oncology. It is not just a database or a tool; it is an expandable knowledge network supporting the import and standardization of genomic and clinical data from cancer research programs. The GDC contains NCI-generated data from some of the largest and most comprehensive cancer genomic datasets, including The Cancer Genome Atlas (TCGA) and Therapeutically Applicable Research to Generate Effective Therapies (TARGET). For the first time, these datasets have been harmonized using a common set of bioinformatics pipelines, so that the data can be directly compared. As a growing knowledge system for cancer, the GDC also enables researchers to submit data, and harmonizes these data for import into the GDC. As more researchers add clinical and genomic data to the GDC, it will become an even more powerful tool for making discoveries about the molecular basis of cancer that may lead to better care for patients.</p>
</blockquote>
<p>The <a href="https://gdc.cancer.gov/developers/gdc-data-model/gdc-data-model-components" class="external-link">data model for the GDC is complex</a>, but it worth a quick overview and a graphical representation is included here.</p>
<div class="figure">
<img src="all_nodes_040318.png" alt=""><p class="caption">The data model is encoded as a so-called property graph. Nodes represent entities such as Projects, Cases, Diagnoses, Files (various kinds), and Annotations. The relationships between these entities are maintained as edges. Both nodes and edges may have Properties that supply instance details.</p>
</div>
<p>The GDC API exposes these nodes and edges in a somewhat simplified set of <a href="https://en.wikipedia.org/wiki/Representational_state_transfer" class="external-link">RESTful</a> endpoints.</p>
</div>
<div class="section level2">
<h2 id="quickstart">Quickstart<a class="anchor" aria-label="anchor" href="#quickstart"></a>
</h2>
<p>This quickstart section is just meant to show basic functionality. More details of functionality are included further on in this vignette and in function-specific help.</p>
<p>This software is available at Bioconductor.org and can be downloaded via <code><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">BiocManager::install</a></code>.</p>
<p>To report bugs or problems, either <a href="https://github.com/Bioconductor/GenomicDataCommons/issues" class="external-link">submit a new issue</a> or submit a <code>bug.report(package='GenomicDataCommons')</code> from within R (which will redirect you to the new issue on GitHub).</p>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>Installation can be achieved via Bioconductor’s <code>BiocManager</code> package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">'GenomicDataCommons'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicDataCommons" class="external-link">GenomicDataCommons</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="check-connectivity-and-status">Check connectivity and status<a class="anchor" aria-label="anchor" href="#check-connectivity-and-status"></a>
</h3>
<p>The <em><a href="https://bioconductor.org/packages/3.15/GenomicDataCommons" class="external-link">GenomicDataCommons</a></em> package relies on having network connectivity. In addition, the NCI GDC API must also be operational and not under maintenance. Checking <code>status</code> can be used to check this connectivity and functionality.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="../reference/status.html">status</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $commit</span></span>
<span><span class="co">## [1] "4dd3680528a19ed33cfc83c7d049426c97bb903b"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $data_release</span></span>
<span><span class="co">## [1] "Data Release 34.0 - July 27, 2022"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $status</span></span>
<span><span class="co">## [1] "OK"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $tag</span></span>
<span><span class="co">## [1] "3.0.0"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $version</span></span>
<span><span class="co">## [1] 1</span></span></code></pre>
<p>And to check the status in code:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span><span class="op">(</span><span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="../reference/status.html">status</a></span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">status</span><span class="op">==</span><span class="st">"OK"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="find-data">Find data<a class="anchor" aria-label="anchor" href="#find-data"></a>
</h3>
<p>The following code builds a <code>manifest</code> that can be used to guide the download of raw data. Here, filtering finds gene expression files quantified as raw counts using <code>STAR</code> from ovarian cancer patients.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ge_manifest</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query.html">files</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/filtering.html">filter</a></span><span class="op">(</span> <span class="va">cases.project.project_id</span> <span class="op">==</span> <span class="st">'TCGA-OV'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="../reference/filtering.html">filter</a></span><span class="op">(</span> <span class="va">type</span> <span class="op">==</span> <span class="st">'gene_expression'</span> <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/filtering.html">filter</a></span><span class="op">(</span> <span class="va">analysis.workflow_type</span> <span class="op">==</span> <span class="st">'STAR - Counts'</span><span class="op">)</span>  <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/manifest.html">manifest</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ge_manifest</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 5</span></span></span>
<span><span class="co">##   id                                   filename               md5     size state</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> 7c69529f-2273-4dc4-b213-e84924d78bea d6472bd0-b4e2-4ed1-a8… 19d5… 4.25<span style="color: #949494;">e</span>6 rele…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> 0eff4634-f8c4-4db9-8a7c-331b21689bae 42165baf-b32c-4fc4-8b… d89d… 4.26<span style="color: #949494;">e</span>6 rele…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> 7d74b4c5-6391-4b3e-95a3-020ea0869e86 accf08d4-a784-4908-83… fb83… 3.11<span style="color: #949494;">e</span>6 rele…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> dc2aeea4-3cd0-4623-92f4-bbbc962851cc 8ab508b9-2993-4e66-b8… 2623… 4.61<span style="color: #949494;">e</span>6 rele…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> 0cf852be-d2e3-4fde-bba8-c93efae2961a 93831282-1dd1-49a3-ac… 0507… 4.27<span style="color: #949494;">e</span>6 rele…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> d33ad23e-2413-419c-8b0b-93ed00583033 bffe7439-0f9b-422f-b0… 1bba… 4.47<span style="color: #949494;">e</span>6 rele…</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="download-data">Download data<a class="anchor" aria-label="anchor" href="#download-data"></a>
</h3>
<p>After the 762 gene expression files specified in the query above. Using multiple processes to do the download very significantly speeds up the transfer in many cases. On a standard 1Gb connection, the following completes in about 30 seconds. The first time the data are downloaded, R will ask to create a cache directory (see <code><a href="../reference/gdc_cache.html">?gdc_cache</a></code> for details of setting and interacting with the cache). Resulting downloaded files will be stored in the cache directory. Future access to the same files will be directly from the cache, alleviating multiple downloads.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fnames</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">ge_manifest</span><span class="op">$</span><span class="va">id</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span>, <span class="va">gdcdata</span><span class="op">)</span></span></code></pre></div>
<p>If the download had included controlled-access data, the download above would have needed to include a <code>token</code>. Details are available in <a href="#authentication">the authentication section below</a>.</p>
</div>
<div class="section level3">
<h3 id="metadata-queries">Metadata queries<a class="anchor" aria-label="anchor" href="#metadata-queries"></a>
</h3>
<div class="section level4">
<h4 id="clinical-data">Clinical data<a class="anchor" aria-label="anchor" href="#clinical-data"></a>
</h4>
<p>Accessing clinical data is a very common task. Given a set of <code>case_ids</code>, the <code><a href="../reference/gdc_clinical.html">gdc_clinical()</a></code> function will return a list of four <code>tibble</code>s.</p>
<ul>
<li>demographic</li>
<li>diagnoses</li>
<li>exposures</li>
<li>main</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">case_ids</span> <span class="op">=</span> <span class="fu"><a href="../reference/query.html">cases</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/results.html">results</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">10</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/ids.html">ids</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">clindat</span> <span class="op">=</span> <span class="fu"><a href="../reference/gdc_clinical.html">gdc_clinical</a></span><span class="op">(</span><span class="va">case_ids</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">clindat</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "demographic" "diagnoses"   "exposures"   "main"</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">clindat</span><span class="op">[[</span><span class="st">"main"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 11</span></span></span>
<span><span class="co">##   id     lost_…¹ disea…² days_…³ submi…⁴ creat…⁵ prima…⁶ updat…⁷ case_id index…⁸</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> 2525b… <span style="color: #BB0000;">NA</span>      Ductal… <span style="color: #BB0000;">NA</span>      C3N-03… 2019-0… Pancre… 2021-0… 2525bf… Diagno…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> 12650… <span style="color: #BB0000;">NA</span>      Gliomas <span style="color: #BB0000;">NA</span>      C3N-01… 2018-0… Brain   2020-0… 126507… Diagno…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> c43ac… <span style="color: #BB0000;">NA</span>      Squamo… <span style="color: #BB0000;">NA</span>      C3N-03… 2019-0… Other … 2020-1… c43ac4… Diagno…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> a59a9… <span style="color: #BB0000;">NA</span>      Squamo… <span style="color: #BB0000;">NA</span>      C3N-02… 2019-0… Other … 2020-1… a59a90… Diagno…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> 59122… <span style="color: #BB0000;">NA</span>      Adenom… <span style="color: #BB0000;">NA</span>      C3L-03… 2019-0… Bronch… 2022-0… 59122a… Diagno…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> 4447a… <span style="color: #BB0000;">NA</span>      Gliomas <span style="color: #BB0000;">NA</span>      C3L-03… 2019-0… Brain   2020-0… 4447a9… Diagno…</span></span>
<span><span class="co">## <span style="color: #949494;"># … with 1 more variable: state &lt;chr&gt;, and abbreviated variable names</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   ¹​lost_to_followup, ²​disease_type, ³​days_to_lost_to_followup, ⁴​submitter_id,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   ⁵​created_datetime, ⁶​primary_site, ⁷​updated_datetime, ⁸​index_date</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ Use `colnames()` to see all variable names</span></span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">clindat</span><span class="op">[[</span><span class="st">"diagnoses"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 82</span></span></span>
<span><span class="co">##   case_id    irs_s…¹ iss_s…² ajcc_…³ ann_a…⁴ created_datetime    ennek…⁵ inrg_…⁶</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dttm&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> 2525bfef-… <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      Stage … <span style="color: #BB0000;">NA</span>      2019-07-22 <span style="color: #949494;">11:40:02</span> <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> 126507c3-… <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      Not Re… <span style="color: #BB0000;">NA</span>      2018-12-03 <span style="color: #949494;">18:05:16</span> <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> c43ac461-… <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      Stage … <span style="color: #BB0000;">NA</span>      2019-03-14 <span style="color: #949494;">15:37:34</span> <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> a59a90d9-… <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      Not Re… <span style="color: #BB0000;">NA</span>      2019-03-14 <span style="color: #949494;">15:37:34</span> <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> 59122a43-… <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      Not Re… <span style="color: #BB0000;">NA</span>      2019-07-22 <span style="color: #949494;">11:40:02</span> <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> 4447a969-… <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      Not Re… <span style="color: #BB0000;">NA</span>      2019-05-07 <span style="color: #949494;">12:41:33</span> <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">## <span style="color: #949494;"># … with 74 more variables: enneking_msts_metastasis &lt;lgl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   tissue_or_organ_of_origin &lt;chr&gt;, age_at_diagnosis &lt;int&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   esophageal_columnar_dysplasia_degree &lt;lgl&gt;, cog_liver_stage &lt;lgl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   child_pugh_classification &lt;lgl&gt;, metastasis_at_diagnosis_site &lt;lgl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   state &lt;chr&gt;, prior_treatment &lt;lgl&gt;, cog_rhabdomyosarcoma_risk_group &lt;lgl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   days_to_last_known_disease_status &lt;dbl&gt;, primary_gleason_grade &lt;lgl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   inpc_grade &lt;lgl&gt;, ajcc_pathologic_t &lt;chr&gt;, morphology &lt;chr&gt;, …</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ Use `colnames()` to see all variable names</span></span></span></code></pre>
</div>
<div class="section level4">
<h4 id="general-metadata-queries">General metadata queries<a class="anchor" aria-label="anchor" href="#general-metadata-queries"></a>
</h4>
<p>The <em><a href="https://bioconductor.org/packages/3.15/GenomicDataCommons" class="external-link">GenomicDataCommons</a></em> package can access the significant clinical, demographic, biospecimen, and annotation information contained in the NCI GDC. The <code><a href="../reference/gdc_clinical.html">gdc_clinical()</a></code> function will often be all that is needed, but the API and <em><a href="https://bioconductor.org/packages/3.15/GenomicDataCommons" class="external-link">GenomicDataCommons</a></em> package make much flexibility if fine-tuning is required.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expands</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"diagnoses"</span>,<span class="st">"annotations"</span>,</span>
<span>             <span class="st">"demographic"</span>,<span class="st">"exposures"</span><span class="op">)</span></span>
<span><span class="va">clinResults</span> <span class="op">=</span> <span class="fu"><a href="../reference/query.html">cases</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="../reference/select.html">select</a></span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="../reference/expand.html">expand</a></span><span class="op">(</span><span class="va">expands</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/results.html">results</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">50</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">clinResults</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,list.len<span class="op">=</span><span class="fl">6</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  chr [1:50] "2525bfef-6962-4b7f-8e80-6186400ce624" ...</span></span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># or listviewer::jsonedit(clinResults)</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="basic-design">Basic design<a class="anchor" aria-label="anchor" href="#basic-design"></a>
</h2>
<p>This package design is meant to have some similarities to the “hadleyverse” approach of dplyr. Roughly, the functionality for finding and accessing files and metadata can be divided into:</p>
<ol style="list-style-type: decimal">
<li>Simple query constructors based on GDC API endpoints.</li>
<li>A set of verbs that when applied, adjust filtering, field selection, and faceting (fields for aggregation) and result in a new query object (an endomorphism)</li>
<li>A set of verbs that take a query and return results from the GDC</li>
</ol>
<p>In addition, there are exhiliary functions for asking the GDC API for information about available and default fields, slicing BAM files, and downloading actual data files. Here is an overview of functionality<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;See individual function and methods documentation for specific details.&lt;/p&gt;"><sup>1</sup></a>.</p>
<ul>
<li>Creating a query
<ul>
<li><code><a href="../reference/query.html">projects()</a></code></li>
<li><code><a href="../reference/query.html">cases()</a></code></li>
<li><code><a href="../reference/query.html">files()</a></code></li>
<li><code><a href="../reference/query.html">annotations()</a></code></li>
</ul>
</li>
<li>Manipulating a query
<ul>
<li><code><a href="../reference/filtering.html">filter()</a></code></li>
<li><code><a href="../reference/faceting.html">facet()</a></code></li>
<li><code><a href="../reference/select.html">select()</a></code></li>
</ul>
</li>
<li>Introspection on the GDC API fields
<ul>
<li><code><a href="../reference/mapping.html">mapping()</a></code></li>
<li><code><a href="../reference/available_fields.html">available_fields()</a></code></li>
<li><code><a href="../reference/default_fields.html">default_fields()</a></code></li>
<li><code><a href="../reference/grep_fields.html">grep_fields()</a></code></li>
<li><code><a href="../reference/available_values.html">available_values()</a></code></li>
<li><code><a href="../reference/available_expand.html">available_expand()</a></code></li>
</ul>
</li>
<li>Executing an API call to retrieve query results
<ul>
<li><code><a href="../reference/results.html">results()</a></code></li>
<li><code><a href="../reference/count.html">count()</a></code></li>
<li><code><a href="../reference/response.html">response()</a></code></li>
</ul>
</li>
<li>Raw data file downloads
<ul>
<li><code><a href="../reference/gdcdata.html">gdcdata()</a></code></li>
<li><code><a href="../reference/transfer.html">transfer()</a></code></li>
<li><code><a href="../reference/gdc_client.html">gdc_client()</a></code></li>
</ul>
</li>
<li>Summarizing and aggregating field values (faceting)
<ul>
<li><code><a href="../reference/aggregations.html">aggregations()</a></code></li>
</ul>
</li>
<li>Authentication
<ul>
<li><code><a href="../reference/gdc_token.html">gdc_token()</a></code></li>
</ul>
</li>
<li>BAM file slicing
<ul>
<li><code><a href="../reference/slicing.html">slicing()</a></code></li>
</ul>
</li>
</ul>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<p>There are two main classes of operations when working with the NCI GDC.</p>
<ol style="list-style-type: decimal">
<li>
<a href="#querying-metadata">Querying metadata and finding data files</a> (e.g., finding all gene expression quantifications data files for all colon cancer patients).</li>
<li>
<a href="#datafile-access-and-download">Transferring raw or processed data</a> from the GDC to another computer (e.g., downloading raw or processed data)</li>
</ol>
<p>Both classes of operation are reviewed in detail in the following sections.</p>
<div class="section level3">
<h3 id="querying-metadata">Querying metadata<a class="anchor" aria-label="anchor" href="#querying-metadata"></a>
</h3>
<p>Vast amounts of metadata about cases (patients, basically), files, projects, and so-called annotations are available via the NCI GDC API. Typically, one will want to query metadata to either focus in on a set of files for download or transfer <em>or</em> to perform so-called aggregations (pivot-tables, facets, similar to the R <code><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table()</a></code> functionality).</p>
<p>Querying metadata starts with <a href="#creating-a-query">creating a “blank” query</a>. One will often then want to <a href="#filtering"><code>filter</code></a> the query to limit results prior to <a href="#retrieving-results">retrieving results</a>. The GenomicDataCommons package has <a href="#fields-and-values">helper functions for listing fields</a> that are available for filtering.</p>
<p>In addition to fetching results, the GDC API allows <a href="#facets-and-aggregation">faceting, or aggregating,</a>, useful for compiling reports, generating dashboards, or building user interfaces to GDC data (see GDC web query interface for a non-R-based example).</p>
<div class="section level4">
<h4 id="creating-a-query">Creating a query<a class="anchor" aria-label="anchor" href="#creating-a-query"></a>
</h4>
<p>A query of the GDC starts its life in R. Queries follow the four metadata endpoints available at the GDC. In particular, there are four convenience functions that each create <code>GDCQuery</code> objects (actually, specific subclasses of <code>GDCQuery</code>):</p>
<ul>
<li><code><a href="../reference/query.html">projects()</a></code></li>
<li><code><a href="../reference/query.html">cases()</a></code></li>
<li><code><a href="../reference/query.html">files()</a></code></li>
<li><code><a href="../reference/query.html">annotations()</a></code></li>
</ul>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pquery</span> <span class="op">=</span> <span class="fu"><a href="../reference/query.html">projects</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>The <code>pquery</code> object is now an object of (S3) class, <code>GDCQuery</code> (and <code>gdc_projects</code> and <code>list</code>). The object contains the following elements:</p>
<ul>
<li>fields: This is a character vector of the fields that will be returned when we <a href="#retrieving-results">retrieve data</a>. If no fields are specified to, for example, the <code><a href="../reference/query.html">projects()</a></code> function, the default fields from the GDC are used (see <code><a href="../reference/default_fields.html">default_fields()</a></code>)</li>
<li>filters: This will contain results after calling the <a href="#filtering"><code>filter()</code> method</a> and will be used to filter results on <a href="#retrieving-results">retrieval</a>.</li>
<li>facets: A character vector of field names that will be used for <a href="#facets-and-aggregation">aggregating data</a> in a call to <code><a href="../reference/aggregations.html">aggregations()</a></code>.</li>
<li>archive: One of either “default” or <a href="https://gdc-portal.nci.nih.gov/legacy-archive/" class="external-link">“legacy”</a>.</li>
<li>token: A character(1) token from the GDC. See <a href="#authentication">the authentication section</a> for details, but note that, in general, the token is not necessary for metadata query and retrieval, only for actual data download.</li>
</ul>
<p>Looking at the actual object (get used to using <code><a href="https://rdrr.io/r/utils/str.html" class="external-link">str()</a></code>!), note that the query contains no results.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">pquery</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of 5</span></span>
<span><span class="co">##  $ fields : chr [1:10] "dbgap_accession_number" "disease_type" "intended_release_date" "name" ...</span></span>
<span><span class="co">##  $ filters: NULL</span></span>
<span><span class="co">##  $ facets : NULL</span></span>
<span><span class="co">##  $ legacy : logi FALSE</span></span>
<span><span class="co">##  $ expand : NULL</span></span>
<span><span class="co">##  - attr(*, "class")= chr [1:3] "gdc_projects" "GDCQuery" "list"</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="retrieving-results">Retrieving results<a class="anchor" aria-label="anchor" href="#retrieving-results"></a>
</h4>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#size-and-from" class="external-link">[ GDC pagination documentation ]</a></p>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#sort" class="external-link">[ GDC sorting documentation ]</a></p>
<p>With a query object available, the next step is to retrieve results from the GDC. The GenomicDataCommons package. The most basic type of results we can get is a simple <code><a href="../reference/count.html">count()</a></code> of records available that satisfy the filter criteria. Note that we have not set any filters, so a <code><a href="../reference/count.html">count()</a></code> here will represent all the project records publicly available at the GDC in the “default” archive"</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pcount</span> <span class="op">=</span> <span class="fu"><a href="../reference/count.html">count</a></span><span class="op">(</span><span class="va">pquery</span><span class="op">)</span></span>
<span><span class="co"># or</span></span>
<span><span class="va">pcount</span> <span class="op">=</span> <span class="va">pquery</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/count.html">count</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">pcount</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 72</span></span></code></pre>
<p>The <code><a href="../reference/results.html">results()</a></code> method will fetch actual results.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">presults</span> <span class="op">=</span> <span class="va">pquery</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/results.html">results</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>These results are returned from the GDC in <a href="http://www.json.org/" class="external-link">JSON</a> format and converted into a (potentially nested) list in R. The <code><a href="https://rdrr.io/r/utils/str.html" class="external-link">str()</a></code> method is useful for taking a quick glimpse of the data.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">presults</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of 9</span></span>
<span><span class="co">##  $ id                    : chr [1:10] "CPTAC-3" "GENIE-GRCC" "GENIE-DFCI" "GENIE-NKI" ...</span></span>
<span><span class="co">##  $ primary_site          :List of 10</span></span>
<span><span class="co">##   ..$ CPTAC-3   : chr [1:6] "Bronchus and lung" "Other and ill-defined sites" "Brain" "Kidney" ...</span></span>
<span><span class="co">##   ..$ GENIE-GRCC: chr [1:45] "Other and ill-defined sites in lip, oral cavity and pharynx" "Uterus, NOS" "Rectum" "Ovary" ...</span></span>
<span><span class="co">##   ..$ GENIE-DFCI: chr [1:49] "Eye and adnexa" "Other and ill-defined sites in lip, oral cavity and pharynx" "Uterus, NOS" "Rectum" ...</span></span>
<span><span class="co">##   ..$ GENIE-NKI : chr [1:42] "Eye and adnexa" "Other and ill-defined sites in lip, oral cavity and pharynx" "Uterus, NOS" "Rectum" ...</span></span>
<span><span class="co">##   ..$ GENIE-VICC: chr [1:46] "Bronchus and lung" "Adrenal gland" "Gallbladder" "Esophagus" ...</span></span>
<span><span class="co">##   ..$ GENIE-UHN : chr [1:42] "Eye and adnexa" "Other and ill-defined sites in lip, oral cavity and pharynx" "Uterus, NOS" "Rectum" ...</span></span>
<span><span class="co">##   ..$ GENIE-MDA : chr [1:42] "Eye and adnexa" "Uterus, NOS" "Ovary" "Other and unspecified urinary organs" ...</span></span>
<span><span class="co">##   ..$ GENIE-MSK : chr [1:49] "Eye and adnexa" "Other and ill-defined sites in lip, oral cavity and pharynx" "Uterus, NOS" "Rectum" ...</span></span>
<span><span class="co">##   ..$ GENIE-JHU : chr [1:33] "Eye and adnexa" "Uterus, NOS" "Rectum" "Ovary" ...</span></span>
<span><span class="co">##   ..$ FM-AD     : chr [1:42] "Bronchus and lung" "Esophagus" "Cervix uteri" "Other and unspecified female genital organs" ...</span></span>
<span><span class="co">##  $ dbgap_accession_number: chr [1:10] "phs001287" NA NA NA ...</span></span>
<span><span class="co">##  $ project_id            : chr [1:10] "CPTAC-3" "GENIE-GRCC" "GENIE-DFCI" "GENIE-NKI" ...</span></span>
<span><span class="co">##  $ disease_type          :List of 10</span></span>
<span><span class="co">##   ..$ CPTAC-3   : chr [1:5] "Ductal and Lobular Neoplasms" "Gliomas" "Adenomas and Adenocarcinomas" "Squamous Cell Neoplasms" ...</span></span>
<span><span class="co">##   ..$ GENIE-GRCC: chr [1:32] "Osseous and Chondromatous Neoplasms" "Synovial-like Neoplasms" "Fibromatous Neoplasms" "Myomatous Neoplasms" ...</span></span>
<span><span class="co">##   ..$ GENIE-DFCI: chr [1:52] "Osseous and Chondromatous Neoplasms" "Other Leukemias" "Synovial-like Neoplasms" "Lymphoid Leukemias" ...</span></span>
<span><span class="co">##   ..$ GENIE-NKI : chr [1:23] "Synovial-like Neoplasms" "Fibromatous Neoplasms" "Myomatous Neoplasms" "Transitional Cell Papillomas and Carcinomas" ...</span></span>
<span><span class="co">##   ..$ GENIE-VICC: chr [1:43] "Neoplasms, NOS" "Adnexal and Skin Appendage Neoplasms" "Squamous Cell Neoplasms" "Gliomas" ...</span></span>
<span><span class="co">##   ..$ GENIE-UHN : chr [1:39] "Other Leukemias" "Osseous and Chondromatous Neoplasms" "Synovial-like Neoplasms" "Lymphoid Leukemias" ...</span></span>
<span><span class="co">##   ..$ GENIE-MDA : chr [1:34] "Osseous and Chondromatous Neoplasms" "Synovial-like Neoplasms" "Fibromatous Neoplasms" "Myomatous Neoplasms" ...</span></span>
<span><span class="co">##   ..$ GENIE-MSK : chr [1:49] "Osseous and Chondromatous Neoplasms" "Synovial-like Neoplasms" "Lymphoid Leukemias" "Fibromatous Neoplasms" ...</span></span>
<span><span class="co">##   ..$ GENIE-JHU : chr [1:33] "Osseous and Chondromatous Neoplasms" "Other Leukemias" "Synovial-like Neoplasms" "Lymphoid Leukemias" ...</span></span>
<span><span class="co">##   ..$ FM-AD     : chr [1:23] "Gliomas" "Acinar Cell Neoplasms" "Specialized Gonadal Neoplasms" "Miscellaneous Tumors" ...</span></span>
<span><span class="co">##  $ name                  : chr [1:10] "CPTAC-Brain, Head and Neck, Kidney, Lung, Pancreas, Uterus" "AACR Project GENIE - Contributed by Institut Gustave Roussy" "AACR Project GENIE - Contributed by Dana-Farber Cancer Institute" "AACR Project GENIE - Contributed by Netherlands Cancer Institute" ...</span></span>
<span><span class="co">##  $ releasable            : logi [1:10] TRUE TRUE TRUE TRUE TRUE TRUE ...</span></span>
<span><span class="co">##  $ state                 : chr [1:10] "open" "open" "open" "open" ...</span></span>
<span><span class="co">##  $ released              : logi [1:10] TRUE TRUE TRUE TRUE TRUE TRUE ...</span></span>
<span><span class="co">##  - attr(*, "row.names")= int [1:10] 1 2 3 4 5 6 7 8 9 10</span></span>
<span><span class="co">##  - attr(*, "class")= chr [1:3] "GDCprojectsResults" "GDCResults" "list"</span></span></code></pre>
<p>A default of only 10 records are returned. We can use the <code>size</code> and <code>from</code> arguments to <code><a href="../reference/results.html">results()</a></code> to either page through results or to change the number of results. Finally, there is a convenience method, <code><a href="../reference/results_all.html">results_all()</a></code> that will simply fetch all the available results given a query. Note that <code><a href="../reference/results_all.html">results_all()</a></code> may take a long time and return HUGE result sets if not used carefully. Use of a combination of <code><a href="../reference/count.html">count()</a></code> and <code><a href="../reference/results.html">results()</a></code> to get a sense of the expected data size is probably warranted before calling <code><a href="../reference/results_all.html">results_all()</a></code></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="../reference/ids.html">ids</a></span><span class="op">(</span><span class="va">presults</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 10</span></span></code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">presults</span> <span class="op">=</span> <span class="va">pquery</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/results_all.html">results_all</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="../reference/ids.html">ids</a></span><span class="op">(</span><span class="va">presults</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 72</span></span></code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># includes all records</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="../reference/ids.html">ids</a></span><span class="op">(</span><span class="va">presults</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="../reference/count.html">count</a></span><span class="op">(</span><span class="va">pquery</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE</span></span></code></pre>
<p>Extracting subsets of results or manipulating the results into a more conventional R data structure is not easily generalizable. However, the <a href="https://github.com/hadley/purrr" class="external-link">purrr</a>, <a href="https://renkun.me/rlist/" class="external-link">rlist</a>, and <a href="https://cran.r-project.org/web/packages/data.tree/vignettes/data.tree.html" class="external-link">data.tree</a> packages are all potentially of interest for manipulating complex, nested list structures. For viewing the results in an interactive viewer, consider the <a href="https://github.com/timelyportfolio/listviewer" class="external-link">listviewer</a> package.</p>
</div>
<div class="section level4">
<h4 id="fields-and-values">Fields and Values<a class="anchor" aria-label="anchor" href="#fields-and-values"></a>
</h4>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#fields" class="external-link">[ GDC <code>fields</code> documentation ]</a></p>
<p>Central to querying and retrieving data from the GDC is the ability to specify which fields to return, filtering by fields and values, and faceting or aggregating. The GenomicDataCommons package includes two simple functions, <code><a href="../reference/available_fields.html">available_fields()</a></code> and <code><a href="../reference/default_fields.html">default_fields()</a></code>. Each can operate on a character(1) endpoint name (“cases”, “files”, “annotations”, or “projects”) or a <code>GDCQuery</code> object.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/default_fields.html">default_fields</a></span><span class="op">(</span><span class="st">'files'</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "access"                         "acl"                           </span></span>
<span><span class="co">##  [3] "average_base_quality"           "average_insert_size"           </span></span>
<span><span class="co">##  [5] "average_read_length"            "channel"                       </span></span>
<span><span class="co">##  [7] "chip_id"                        "chip_position"                 </span></span>
<span><span class="co">##  [9] "contamination"                  "contamination_error"           </span></span>
<span><span class="co">## [11] "created_datetime"               "data_category"                 </span></span>
<span><span class="co">## [13] "data_format"                    "data_type"                     </span></span>
<span><span class="co">## [15] "error_type"                     "experimental_strategy"         </span></span>
<span><span class="co">## [17] "file_autocomplete"              "file_id"                       </span></span>
<span><span class="co">## [19] "file_name"                      "file_size"                     </span></span>
<span><span class="co">## [21] "imaging_date"                   "magnification"                 </span></span>
<span><span class="co">## [23] "md5sum"                         "mean_coverage"                 </span></span>
<span><span class="co">## [25] "msi_score"                      "msi_status"                    </span></span>
<span><span class="co">## [27] "pairs_on_diff_chr"              "plate_name"                    </span></span>
<span><span class="co">## [29] "plate_well"                     "platform"                      </span></span>
<span><span class="co">## [31] "proc_internal"                  "proportion_base_mismatch"      </span></span>
<span><span class="co">## [33] "proportion_coverage_10x"        "proportion_coverage_10X"       </span></span>
<span><span class="co">## [35] "proportion_coverage_30x"        "proportion_coverage_30X"       </span></span>
<span><span class="co">## [37] "proportion_reads_duplicated"    "proportion_reads_mapped"       </span></span>
<span><span class="co">## [39] "proportion_targets_no_coverage" "read_pair_number"              </span></span>
<span><span class="co">## [41] "revision"                       "stain_type"                    </span></span>
<span><span class="co">## [43] "state"                          "state_comment"                 </span></span>
<span><span class="co">## [45] "submitter_id"                   "tags"                          </span></span>
<span><span class="co">## [47] "total_reads"                    "tumor_ploidy"                  </span></span>
<span><span class="co">## [49] "tumor_purity"                   "type"                          </span></span>
<span><span class="co">## [51] "updated_datetime"</span></span></code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The number of fields available for files endpoint</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="../reference/available_fields.html">available_fields</a></span><span class="op">(</span><span class="st">'files'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 1017</span></span></code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The first few fields available for files endpoint</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/available_fields.html">available_fields</a></span><span class="op">(</span><span class="st">'files'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "access"                      "acl"                        </span></span>
<span><span class="co">## [3] "analysis.analysis_id"        "analysis.analysis_type"     </span></span>
<span><span class="co">## [5] "analysis.created_datetime"   "analysis.input_files.access"</span></span></code></pre>
<p>The fields to be returned by a query can be specified following a similar paradigm to that of the dplyr package. The <code><a href="../reference/select.html">select()</a></code> function is a verb that resets the fields slot of a <code>GDCQuery</code>; note that this is not quite analogous to the dplyr <code><a href="../reference/select.html">select()</a></code> verb that limits from already-present fields. We <em>completely replace</em> the fields when using <code><a href="../reference/select.html">select()</a></code> on a <code>GDCQuery</code>.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Default fields here</span></span>
<span><span class="va">qcases</span> <span class="op">=</span> <span class="fu"><a href="../reference/query.html">cases</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">qcases</span><span class="op">$</span><span class="va">fields</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "aliquot_ids"              "analyte_ids"             </span></span>
<span><span class="co">##  [3] "case_autocomplete"        "case_id"                 </span></span>
<span><span class="co">##  [5] "consent_type"             "created_datetime"        </span></span>
<span><span class="co">##  [7] "days_to_consent"          "days_to_lost_to_followup"</span></span>
<span><span class="co">##  [9] "diagnosis_ids"            "disease_type"            </span></span>
<span><span class="co">## [11] "index_date"               "lost_to_followup"        </span></span>
<span><span class="co">## [13] "portion_ids"              "primary_site"            </span></span>
<span><span class="co">## [15] "sample_ids"               "slide_ids"               </span></span>
<span><span class="co">## [17] "state"                    "submitter_aliquot_ids"   </span></span>
<span><span class="co">## [19] "submitter_analyte_ids"    "submitter_diagnosis_ids" </span></span>
<span><span class="co">## [21] "submitter_id"             "submitter_portion_ids"   </span></span>
<span><span class="co">## [23] "submitter_sample_ids"     "submitter_slide_ids"     </span></span>
<span><span class="co">## [25] "updated_datetime"</span></span></code></pre>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># set up query to use ALL available fields</span></span>
<span><span class="co"># Note that checking of fields is done by select()</span></span>
<span><span class="va">qcases</span> <span class="op">=</span> <span class="fu"><a href="../reference/query.html">cases</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="../reference/select.html">select</a></span><span class="op">(</span><span class="fu"><a href="../reference/available_fields.html">available_fields</a></span><span class="op">(</span><span class="st">'cases'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">qcases</span><span class="op">$</span><span class="va">fields</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "case_id"                       "aliquot_ids"                  </span></span>
<span><span class="co">## [3] "analyte_ids"                   "annotations.annotation_id"    </span></span>
<span><span class="co">## [5] "annotations.case_id"           "annotations.case_submitter_id"</span></span></code></pre>
<p>Finding fields of interest is such a common operation that the GenomicDataCommons includes the <code><a href="../reference/grep_fields.html">grep_fields()</a></code> function. See the appropriate help pages for details.</p>
</div>
<div class="section level4">
<h4 id="facets-and-aggregation">Facets and aggregation<a class="anchor" aria-label="anchor" href="#facets-and-aggregation"></a>
</h4>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#facets" class="external-link">[ GDC <code>facet</code> documentation ]</a></p>
<p>The GDC API offers a feature known as aggregation or faceting. By specifying one or more fields (of appropriate type), the GDC can return to us a count of the number of records matching each potential value. This is similar to the R <code>table</code> method. Multiple fields can be returned at once, but the GDC API does not have a cross-tabulation feature; all aggregations are only on one field at a time. Results of <code>aggregation()</code> calls come back as a list of data.frames (actually, tibbles).</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># total number of files of a specific type</span></span>
<span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="../reference/query.html">files</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/faceting.html">facet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'type'</span>,<span class="st">'data_type'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/aggregations.html">aggregations</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">res</span><span class="op">$</span><span class="va">type</span></span></code></pre></div>
<pre><code><span><span class="co">##    doc_count                           key</span></span>
<span><span class="co">## 1     222083    annotated_somatic_mutation</span></span>
<span><span class="co">## 2     114445                 aligned_reads</span></span>
<span><span class="co">## 3     102544       simple_somatic_mutation</span></span>
<span><span class="co">## 4      74909          structural_variation</span></span>
<span><span class="co">## 5      58881           copy_number_segment</span></span>
<span><span class="co">## 6      46151          copy_number_estimate</span></span>
<span><span class="co">## 7      37654               gene_expression</span></span>
<span><span class="co">## 8      33636   aggregated_somatic_mutation</span></span>
<span><span class="co">## 9      33234              mirna_expression</span></span>
<span><span class="co">## 10     31714      masked_methylation_array</span></span>
<span><span class="co">## 11     30075                   slide_image</span></span>
<span><span class="co">## 12     26001        biospecimen_supplement</span></span>
<span><span class="co">## 13     15857        methylation_beta_value</span></span>
<span><span class="co">## 14     15633       masked_somatic_mutation</span></span>
<span><span class="co">## 15     13133           clinical_supplement</span></span>
<span><span class="co">## 16      7906            protein_expression</span></span>
<span><span class="co">## 17        67              pathology_report</span></span>
<span><span class="co">## 18        54 secondary_expression_analysis</span></span></code></pre>
<p>Using <code><a href="../reference/aggregations.html">aggregations()</a></code> is an also easy way to learn the contents of individual fields and forms the basis for faceted search pages.</p>
</div>
<div class="section level4">
<h4 id="filtering">Filtering<a class="anchor" aria-label="anchor" href="#filtering"></a>
</h4>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#filters-specifying-the-query" class="external-link">[ GDC <code>filtering</code> documentation ]</a></p>
<p>The GenomicDataCommons package uses a form of non-standard evaluation to specify R-like queries that are then translated into an R list. That R list is, upon calling a method that fetches results from the GDC API, translated into the appropriate JSON string. The R expression uses the formula interface as suggested by Hadley Wickham in his <a href="https://cran.r-project.org/web/packages/dplyr/vignettes/nse.html" class="external-link">vignette on non-standard evaluation</a></p>
<blockquote>
<p>It’s best to use a formula because a formula captures both the expression to evaluate and the environment where the evaluation occurs. This is important if the expression is a mixture of variables in a data frame and objects in the local environment [for example].</p>
</blockquote>
<p>For the user, these details will not be too important except to note that a filter expression must begin with a “~”.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qfiles</span> <span class="op">=</span> <span class="fu"><a href="../reference/query.html">files</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">qfiles</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/count.html">count</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># all files</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 863977</span></span></code></pre>
<p>To limit the file type, we can refer back to the <a href="#facets-and-aggregation">section on faceting</a> to see the possible values for the file field “type”. For example, to filter file results to only “gene_expression” files, we simply specify a filter.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qfiles</span> <span class="op">=</span> <span class="fu"><a href="../reference/query.html">files</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/filtering.html">filter</a></span><span class="op">(</span> <span class="va">type</span> <span class="op">==</span> <span class="st">'gene_expression'</span><span class="op">)</span></span>
<span><span class="co"># here is what the filter looks like after translation</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="fu"><a href="../reference/filtering.html">get_filter</a></span><span class="op">(</span><span class="va">qfiles</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of 2</span></span>
<span><span class="co">##  $ op     : 'scalar' chr "="</span></span>
<span><span class="co">##  $ content:List of 2</span></span>
<span><span class="co">##   ..$ field: chr "type"</span></span>
<span><span class="co">##   ..$ value: chr "gene_expression"</span></span></code></pre>
<p>What if we want to create a filter based on the project (‘TCGA-OVCA’, for example)? Well, we have a couple of possible ways to discover available fields. The first is based on base R functionality and some intuition.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">'pro'</span>,<span class="fu"><a href="../reference/available_fields.html">available_fields</a></span><span class="op">(</span><span class="st">'files'</span><span class="op">)</span>,value<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "analysis.input_files.proc_internal"           </span></span>
<span><span class="co">## [2] "analysis.input_files.proportion_base_mismatch"</span></span>
<span><span class="co">## [3] "analysis.input_files.proportion_coverage_10x" </span></span>
<span><span class="co">## [4] "analysis.input_files.proportion_coverage_10X" </span></span>
<span><span class="co">## [5] "analysis.input_files.proportion_coverage_30x" </span></span>
<span><span class="co">## [6] "analysis.input_files.proportion_coverage_30X"</span></span></code></pre>
<p>Interestingly, the project information is “nested” inside the case. We don’t need to know that detail other than to know that we now have a few potential guesses for where our information might be in the files records. We need to know where because we need to construct the appropriate filter.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/query.html">files</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="../reference/faceting.html">facet</a></span><span class="op">(</span><span class="st">'cases.project.project_id'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="../reference/aggregations.html">aggregations</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $cases.project.project_id</span></span>
<span><span class="co">##    doc_count                       key</span></span>
<span><span class="co">## 1      54096                     FM-AD</span></span>
<span><span class="co">## 2      52659                 TCGA-BRCA</span></span>
<span><span class="co">## 3      60269                   CPTAC-3</span></span>
<span><span class="co">## 4      28359                 TCGA-LUAD</span></span>
<span><span class="co">## 5      36470                 GENIE-MSK</span></span>
<span><span class="co">## 6      24402                 TCGA-UCEC</span></span>
<span><span class="co">## 7      25199                 TCGA-HNSC</span></span>
<span><span class="co">## 8      24350                 TCGA-THCA</span></span>
<span><span class="co">## 9      23453                   TCGA-OV</span></span>
<span><span class="co">## 10     25568                 TCGA-LUSC</span></span>
<span><span class="co">## 11     24844                 TCGA-KIRC</span></span>
<span><span class="co">## 12     24724                  TCGA-LGG</span></span>
<span><span class="co">## 13     24040                 TCGA-PRAD</span></span>
<span><span class="co">## 14     22166                 TCGA-COAD</span></span>
<span><span class="co">## 15     30290             MMRF-COMMPASS</span></span>
<span><span class="co">## 16     19507                  TCGA-GBM</span></span>
<span><span class="co">## 17     21583                 TCGA-SKCM</span></span>
<span><span class="co">## 18     21017                 TCGA-STAD</span></span>
<span><span class="co">## 19     28464                GENIE-DFCI</span></span>
<span><span class="co">## 20     19588                 TCGA-BLCA</span></span>
<span><span class="co">## 21     19176             TARGET-ALL-P2</span></span>
<span><span class="co">## 22     18099                 TCGA-LIHC</span></span>
<span><span class="co">## 23     14260                 TCGA-CESC</span></span>
<span><span class="co">## 24     14266                 TCGA-KIRP</span></span>
<span><span class="co">## 25     12369                 TCGA-SARC</span></span>
<span><span class="co">## 26     11690                TARGET-AML</span></span>
<span><span class="co">## 27     16471         BEATAML1.0-COHORT</span></span>
<span><span class="co">## 28      8739                 TCGA-PAAD</span></span>
<span><span class="co">## 29      8719                 TCGA-ESCA</span></span>
<span><span class="co">## 30     11553                 REBC-THYR</span></span>
<span><span class="co">## 31      8420                 TCGA-PCPG</span></span>
<span><span class="co">## 32      7670                 TCGA-READ</span></span>
<span><span class="co">## 33     10228                   CPTAC-2</span></span>
<span><span class="co">## 34      7199                 TCGA-TGCT</span></span>
<span><span class="co">## 35      6015                TARGET-NBL</span></span>
<span><span class="co">## 36      7708                 TCGA-LAML</span></span>
<span><span class="co">## 37      5745                 TCGA-THYM</span></span>
<span><span class="co">## 38      7563                 HCMI-CMDC</span></span>
<span><span class="co">## 39      5559             CGCI-HTMCP-CC</span></span>
<span><span class="co">## 40      6144                   CMI-MBC</span></span>
<span><span class="co">## 41      4152                  TCGA-ACC</span></span>
<span><span class="co">## 42      3691                 TCGA-KICH</span></span>
<span><span class="co">## 43      3913                 TCGA-MESO</span></span>
<span><span class="co">## 44      5286              NCICCR-DLBCL</span></span>
<span><span class="co">## 45      3623                  TCGA-UVM</span></span>
<span><span class="co">## 46      2659                 TARGET-WT</span></span>
<span><span class="co">## 47      3107                 TARGET-OS</span></span>
<span><span class="co">## 48      4000             TARGET-ALL-P3</span></span>
<span><span class="co">## 49      3857                 GENIE-MDA</span></span>
<span><span class="co">## 50      3833                GENIE-VICC</span></span>
<span><span class="co">## 51      2720                  TCGA-UCS</span></span>
<span><span class="co">## 52      3320                 GENIE-JHU</span></span>
<span><span class="co">## 53      2178                 TCGA-DLBC</span></span>
<span><span class="co">## 54      2212                 TCGA-CHOL</span></span>
<span><span class="co">## 55      2632                 GENIE-UHN</span></span>
<span><span class="co">## 56      2139                CGCI-BLGSP</span></span>
<span><span class="co">## 57      1931 EXCEPTIONAL_RESPONDERS-ER</span></span>
<span><span class="co">## 58      1571                 MP2PRT-WT</span></span>
<span><span class="co">## 59      1036                 TARGET-RT</span></span>
<span><span class="co">## 60      1093                WCDT-MCRPC</span></span>
<span><span class="co">## 61      1038                GENIE-GRCC</span></span>
<span><span class="co">## 62       944                   CMI-ASC</span></span>
<span><span class="co">## 63       878                  OHSU-CNL</span></span>
<span><span class="co">## 64       801                 GENIE-NKI</span></span>
<span><span class="co">## 65       758       ORGANOID-PANCREATIC</span></span>
<span><span class="co">## 66       570                   CMI-MPC</span></span>
<span><span class="co">## 67       553               CTSP-DLBCL1</span></span>
<span><span class="co">## 68       339                  TRIO-CRU</span></span>
<span><span class="co">## 69       222     BEATAML1.0-CRENOLANIB</span></span>
<span><span class="co">## 70       163               TARGET-CCSK</span></span>
<span><span class="co">## 71        96             TARGET-ALL-P1</span></span>
<span><span class="co">## 72        21            VAREPOP-APOLLO</span></span></code></pre>
<p>We note that <code>cases.project.project_id</code> looks like it is a good fit. We also note that <code>TCGA-OV</code> is the correct project_id, not <code>TCGA-OVCA</code>. Note that <em>unlike with dplyr and friends, the <code><a href="../reference/filtering.html">filter()</a></code> method here <strong>replaces</strong> the filter and does not build on any previous filters</em>.</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qfiles</span> <span class="op">=</span> <span class="fu"><a href="../reference/query.html">files</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/filtering.html">filter</a></span><span class="op">(</span> <span class="va">cases.project.project_id</span> <span class="op">==</span> <span class="st">'TCGA-OV'</span> <span class="op">&amp;</span> <span class="va">type</span> <span class="op">==</span> <span class="st">'gene_expression'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="fu"><a href="../reference/filtering.html">get_filter</a></span><span class="op">(</span><span class="va">qfiles</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of 2</span></span>
<span><span class="co">##  $ op     : 'scalar' chr "and"</span></span>
<span><span class="co">##  $ content:List of 2</span></span>
<span><span class="co">##   ..$ :List of 2</span></span>
<span><span class="co">##   .. ..$ op     : 'scalar' chr "="</span></span>
<span><span class="co">##   .. ..$ content:List of 2</span></span>
<span><span class="co">##   .. .. ..$ field: chr "cases.project.project_id"</span></span>
<span><span class="co">##   .. .. ..$ value: chr "TCGA-OV"</span></span>
<span><span class="co">##   ..$ :List of 2</span></span>
<span><span class="co">##   .. ..$ op     : 'scalar' chr "="</span></span>
<span><span class="co">##   .. ..$ content:List of 2</span></span>
<span><span class="co">##   .. .. ..$ field: chr "type"</span></span>
<span><span class="co">##   .. .. ..$ value: chr "gene_expression"</span></span></code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qfiles</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/count.html">count</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 762</span></span></code></pre>
<p>Asking for a <code><a href="../reference/count.html">count()</a></code> of results given these new filter criteria gives <code>r qfiles %&gt;% count()</code> results. Filters can be chained (or nested) to accomplish the same effect as multiple <code>&amp;</code> conditionals. The <code><a href="../reference/count.html">count()</a></code> below is equivalent to the <code>&amp;</code> filtering done above.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qfiles2</span> <span class="op">=</span> <span class="fu"><a href="../reference/query.html">files</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/filtering.html">filter</a></span><span class="op">(</span> <span class="va">cases.project.project_id</span> <span class="op">==</span> <span class="st">'TCGA-OV'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="../reference/filtering.html">filter</a></span><span class="op">(</span> <span class="va">type</span> <span class="op">==</span> <span class="st">'gene_expression'</span><span class="op">)</span> </span>
<span><span class="va">qfiles2</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/count.html">count</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 762</span></span></code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">qfiles</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/count.html">count</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="op">(</span><span class="va">qfiles2</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/count.html">count</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="co">#TRUE</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE</span></span></code></pre>
<p>Generating a manifest for bulk downloads is as simple as asking for the manifest from the current query.</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">manifest_df</span> <span class="op">=</span> <span class="va">qfiles</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/manifest.html">manifest</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">manifest_df</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 5</span></span></span>
<span><span class="co">##   id                                   filename               md5     size state</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> 7c69529f-2273-4dc4-b213-e84924d78bea d6472bd0-b4e2-4ed1-a8… 19d5… 4.25<span style="color: #949494;">e</span>6 rele…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> 0eff4634-f8c4-4db9-8a7c-331b21689bae 42165baf-b32c-4fc4-8b… d89d… 4.26<span style="color: #949494;">e</span>6 rele…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> 7d74b4c5-6391-4b3e-95a3-020ea0869e86 accf08d4-a784-4908-83… fb83… 3.11<span style="color: #949494;">e</span>6 rele…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> dc2aeea4-3cd0-4623-92f4-bbbc962851cc 8ab508b9-2993-4e66-b8… 2623… 4.61<span style="color: #949494;">e</span>6 rele…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> 0cf852be-d2e3-4fde-bba8-c93efae2961a 93831282-1dd1-49a3-ac… 0507… 4.27<span style="color: #949494;">e</span>6 rele…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> d33ad23e-2413-419c-8b0b-93ed00583033 bffe7439-0f9b-422f-b0… 1bba… 4.47<span style="color: #949494;">e</span>6 rele…</span></span></code></pre>
<p>Note that we might still not be quite there. Looking at filenames, there are suspiciously named files that might include “FPKM”, “FPKM-UQ”, or “counts”. Another round of <code>grep</code> and <code>available_fields</code>, looking for “type” turned up that the field “analysis.workflow_type” has the appropriate filter criteria.</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qfiles</span> <span class="op">=</span> <span class="fu"><a href="../reference/query.html">files</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/filtering.html">filter</a></span><span class="op">(</span> <span class="op">~</span> <span class="va">cases.project.project_id</span> <span class="op">==</span> <span class="st">'TCGA-OV'</span> <span class="op">&amp;</span></span>
<span>                            <span class="va">type</span> <span class="op">==</span> <span class="st">'gene_expression'</span> <span class="op">&amp;</span></span>
<span>                            <span class="va">analysis.workflow_type</span> <span class="op">==</span> <span class="st">'STAR - Counts'</span><span class="op">)</span></span>
<span><span class="va">manifest_df</span> <span class="op">=</span> <span class="va">qfiles</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/manifest.html">manifest</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">manifest_df</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 762</span></span></code></pre>
<p>The GDC Data Transfer Tool can be used (from R, <code><a href="../reference/transfer.html">transfer()</a></code> or from the command-line) to orchestrate high-performance, restartable transfers of all the files in the manifest. See <a href="bulk-downloads">the bulk downloads section</a> for details.</p>
</div>
</div>
<div class="section level3">
<h3 id="authentication">Authentication<a class="anchor" aria-label="anchor" href="#authentication"></a>
</h3>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#facets" class="external-link">[ GDC authentication documentation ]</a></p>
<p>The GDC offers both “controlled-access” and “open” data. As of this writing, only data stored as files is “controlled-access”; that is, metadata accessible via the GDC is all “open” data and some files are “open” and some are “controlled-access”. Controlled-access data are only available after <a href="https://gdc.cancer.gov/access-data/obtaining-access-controlled-data" class="external-link">going through the process of obtaining access.</a></p>
<p>After controlled-access to one or more datasets has been granted, logging into the GDC web portal will allow you to <a href="https://docs.gdc.cancer.gov/Data_Portal/Users_Guide/Authentication/#gdc-authentication-tokens" class="external-link">access a GDC authentication token</a>, which can be downloaded and then used to access available controlled-access data via the GenomicDataCommons package.</p>
<p>The GenomicDataCommons uses authentication tokens only for downloading data (see <code>transfer</code> and <code>gdcdata</code> documentation). The package includes a helper function, <code>gdc_token</code>, that looks for the token to be stored in one of three ways (resolved in this order):</p>
<ol style="list-style-type: decimal">
<li>As a string stored in the environment variable, <code>GDC_TOKEN</code>
</li>
<li>As a file, stored in the file named by the environment variable, <code>GDC_TOKEN_FILE</code>
</li>
<li>In a file in the user home directory, called <code>.gdc_token</code>
</li>
</ol>
<p>As a concrete example:</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">token</span> <span class="op">=</span> <span class="fu"><a href="../reference/gdc_token.html">gdc_token</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/transfer.html">transfer</a></span><span class="op">(</span><span class="va">...</span>,token<span class="op">=</span><span class="va">token</span><span class="op">)</span></span>
<span><span class="co"># or</span></span>
<span><span class="fu"><a href="../reference/transfer.html">transfer</a></span><span class="op">(</span><span class="va">...</span>,token<span class="op">=</span><span class="fu">get_token</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="datafile-access-and-download">Datafile access and download<a class="anchor" aria-label="anchor" href="#datafile-access-and-download"></a>
</h3>
<div class="section level4">
<h4 id="data-downloads-via-the-gdc-api">Data downloads via the GDC API<a class="anchor" aria-label="anchor" href="#data-downloads-via-the-gdc-api"></a>
</h4>
<p>The <code>gdcdata</code> function takes a character vector of one or more file ids. A simple way of producing such a vector is to produce a <code>manifest</code> data frame and then pass in the first column, which will contain file ids.</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fnames</span> <span class="op">=</span> <span class="fu"><a href="../reference/gdcdata.html">gdcdata</a></span><span class="op">(</span><span class="va">manifest_df</span><span class="op">$</span><span class="va">id</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>,progress<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>Note that for controlled-access data, a GDC <a href="#authentication">authentication token</a> is required. Using the <code>BiocParallel</code> package may be useful for downloading in parallel, particularly for large numbers of smallish files.</p>
</div>
<div class="section level4">
<h4 id="bulk-downloads">Bulk downloads<a class="anchor" aria-label="anchor" href="#bulk-downloads"></a>
</h4>
<p>The bulk download functionality is only efficient (as of v1.2.0 of the GDC Data Transfer Tool) for relatively large files, so use this approach only when transferring BAM files or larger VCF files, for example. Otherwise, consider using the approach shown above, perhaps in parallel.</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Requires gcd_client command-line utility to be isntalled</span></span>
<span><span class="co"># separately. </span></span>
<span><span class="va">fnames</span> <span class="op">=</span> <span class="fu"><a href="../reference/gdcdata.html">gdcdata</a></span><span class="op">(</span><span class="va">manifest_df</span><span class="op">$</span><span class="va">id</span><span class="op">[</span><span class="fl">3</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>, access_method <span class="op">=</span> <span class="st">'client'</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="bam-slicing">BAM slicing<a class="anchor" aria-label="anchor" href="#bam-slicing"></a>
</h4>
</div>
</div>
</div>
<div class="section level2">
<h2 id="use-cases">Use Cases<a class="anchor" aria-label="anchor" href="#use-cases"></a>
</h2>
<div class="section level3">
<h3 id="cases">Cases<a class="anchor" aria-label="anchor" href="#cases"></a>
</h3>
<div class="section level4">
<h4 id="how-many-cases-are-there-per-project_id">How many cases are there per project_id?<a class="anchor" aria-label="anchor" href="#how-many-cases-are-there-per-project_id"></a>
</h4>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="../reference/query.html">cases</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/faceting.html">facet</a></span><span class="op">(</span><span class="st">"project.project_id"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/aggregations.html">aggregations</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $project.project_id</span></span>
<span><span class="co">##    doc_count                       key</span></span>
<span><span class="co">## 1      18004                     FM-AD</span></span>
<span><span class="co">## 2      16824                 GENIE-MSK</span></span>
<span><span class="co">## 3      14232                GENIE-DFCI</span></span>
<span><span class="co">## 4       3857                 GENIE-MDA</span></span>
<span><span class="co">## 5       3320                 GENIE-JHU</span></span>
<span><span class="co">## 6       2632                 GENIE-UHN</span></span>
<span><span class="co">## 7       2145                TARGET-AML</span></span>
<span><span class="co">## 8       2052                GENIE-VICC</span></span>
<span><span class="co">## 9       1587             TARGET-ALL-P2</span></span>
<span><span class="co">## 10      1132                TARGET-NBL</span></span>
<span><span class="co">## 11      1098                 TCGA-BRCA</span></span>
<span><span class="co">## 12      1046                   CPTAC-3</span></span>
<span><span class="co">## 13      1038                GENIE-GRCC</span></span>
<span><span class="co">## 14       995             MMRF-COMMPASS</span></span>
<span><span class="co">## 15       826         BEATAML1.0-COHORT</span></span>
<span><span class="co">## 16       801                 GENIE-NKI</span></span>
<span><span class="co">## 17       652                 TARGET-WT</span></span>
<span><span class="co">## 18       617                  TCGA-GBM</span></span>
<span><span class="co">## 19       608                   TCGA-OV</span></span>
<span><span class="co">## 20       585                 TCGA-LUAD</span></span>
<span><span class="co">## 21       560                 TCGA-UCEC</span></span>
<span><span class="co">## 22       537                 TCGA-KIRC</span></span>
<span><span class="co">## 23       528                 TCGA-HNSC</span></span>
<span><span class="co">## 24       516                  TCGA-LGG</span></span>
<span><span class="co">## 25       507                 TCGA-THCA</span></span>
<span><span class="co">## 26       504                 TCGA-LUSC</span></span>
<span><span class="co">## 27       500                 TCGA-PRAD</span></span>
<span><span class="co">## 28       489              NCICCR-DLBCL</span></span>
<span><span class="co">## 29       470                 TCGA-SKCM</span></span>
<span><span class="co">## 30       461                 TCGA-COAD</span></span>
<span><span class="co">## 31       443                 TCGA-STAD</span></span>
<span><span class="co">## 32       440                 REBC-THYR</span></span>
<span><span class="co">## 33       412                 TCGA-BLCA</span></span>
<span><span class="co">## 34       383                 TARGET-OS</span></span>
<span><span class="co">## 35       377                 TCGA-LIHC</span></span>
<span><span class="co">## 36       342                   CPTAC-2</span></span>
<span><span class="co">## 37       339                  TRIO-CRU</span></span>
<span><span class="co">## 38       307                 TCGA-CESC</span></span>
<span><span class="co">## 39       291                 TCGA-KIRP</span></span>
<span><span class="co">## 40       261                 TCGA-SARC</span></span>
<span><span class="co">## 41       212             CGCI-HTMCP-CC</span></span>
<span><span class="co">## 42       200                   CMI-MBC</span></span>
<span><span class="co">## 43       200                 TCGA-LAML</span></span>
<span><span class="co">## 44       191             TARGET-ALL-P3</span></span>
<span><span class="co">## 45       185                 TCGA-ESCA</span></span>
<span><span class="co">## 46       185                 TCGA-PAAD</span></span>
<span><span class="co">## 47       179                 TCGA-PCPG</span></span>
<span><span class="co">## 48       176                  OHSU-CNL</span></span>
<span><span class="co">## 49       172                 TCGA-READ</span></span>
<span><span class="co">## 50       150                 TCGA-TGCT</span></span>
<span><span class="co">## 51       124                 TCGA-THYM</span></span>
<span><span class="co">## 52       120                CGCI-BLGSP</span></span>
<span><span class="co">## 53       113                 TCGA-KICH</span></span>
<span><span class="co">## 54       109                 HCMI-CMDC</span></span>
<span><span class="co">## 55       101                WCDT-MCRPC</span></span>
<span><span class="co">## 56        92                  TCGA-ACC</span></span>
<span><span class="co">## 57        87                 TCGA-MESO</span></span>
<span><span class="co">## 58        84 EXCEPTIONAL_RESPONDERS-ER</span></span>
<span><span class="co">## 59        80                  TCGA-UVM</span></span>
<span><span class="co">## 60        70       ORGANOID-PANCREATIC</span></span>
<span><span class="co">## 61        69                 TARGET-RT</span></span>
<span><span class="co">## 62        58                 TCGA-DLBC</span></span>
<span><span class="co">## 63        57                  TCGA-UCS</span></span>
<span><span class="co">## 64        56     BEATAML1.0-CRENOLANIB</span></span>
<span><span class="co">## 65        52                 MP2PRT-WT</span></span>
<span><span class="co">## 66        51                 TCGA-CHOL</span></span>
<span><span class="co">## 67        45               CTSP-DLBCL1</span></span>
<span><span class="co">## 68        36                   CMI-ASC</span></span>
<span><span class="co">## 69        30                   CMI-MPC</span></span>
<span><span class="co">## 70        24             TARGET-ALL-P1</span></span>
<span><span class="co">## 71        13               TARGET-CCSK</span></span>
<span><span class="co">## 72         7            VAREPOP-APOLLO</span></span></code></pre>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">project.project_id</span>,<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">key</span>, y <span class="op">=</span> <span class="va">doc_count</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat<span class="op">=</span><span class="st">'identity'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="overview_files/figure-html/casesPerProject-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="how-many-cases-are-included-in-all-target-projects">How many cases are included in all TARGET projects?<a class="anchor" aria-label="anchor" href="#how-many-cases-are-included-in-all-target-projects"></a>
</h4>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/query.html">cases</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/filtering.html">filter</a></span><span class="op">(</span><span class="op">~</span> <span class="va">project.program.name</span><span class="op">==</span><span class="st">'TARGET'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/count.html">count</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 6196</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="how-many-cases-are-included-in-all-tcga-projects">How many cases are included in all TCGA projects?<a class="anchor" aria-label="anchor" href="#how-many-cases-are-included-in-all-tcga-projects"></a>
</h4>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/query.html">cases</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/filtering.html">filter</a></span><span class="op">(</span><span class="op">~</span> <span class="va">project.program.name</span><span class="op">==</span><span class="st">'TCGA'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/count.html">count</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 11315</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="what-is-the-breakdown-of-sample-types-in-tcga-brca">What is the breakdown of sample types in TCGA-BRCA?<a class="anchor" aria-label="anchor" href="#what-is-the-breakdown-of-sample-types-in-tcga-brca"></a>
</h4>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The need to do the "&amp;" here is a requirement of the</span></span>
<span><span class="co"># current version of the GDC API. I have filed a feature</span></span>
<span><span class="co"># request to remove this requirement.</span></span>
<span><span class="va">resp</span> <span class="op">=</span> <span class="fu"><a href="../reference/query.html">cases</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/filtering.html">filter</a></span><span class="op">(</span><span class="op">~</span> <span class="va">project.project_id</span><span class="op">==</span><span class="st">'TCGA-BRCA'</span> <span class="op">&amp;</span></span>
<span>                              <span class="va">project.project_id</span><span class="op">==</span><span class="st">'TCGA-BRCA'</span> <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/faceting.html">facet</a></span><span class="op">(</span><span class="st">'samples.sample_type'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/aggregations.html">aggregations</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">resp</span><span class="op">$</span><span class="va">samples.sample_type</span></span></code></pre></div>
<pre><code><span><span class="co">##   doc_count                  key</span></span>
<span><span class="co">## 1      1098        primary tumor</span></span>
<span><span class="co">## 2      1011 blood derived normal</span></span>
<span><span class="co">## 3       162  solid tissue normal</span></span>
<span><span class="co">## 4         7           metastatic</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="fetch-all-samples-in-tcga-brca-that-use-solid-tissue-as-a-normal-">Fetch all samples in TCGA-BRCA that use “Solid Tissue” as a normal.<a class="anchor" aria-label="anchor" href="#fetch-all-samples-in-tcga-brca-that-use-solid-tissue-as-a-normal-"></a>
</h4>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The need to do the "&amp;" here is a requirement of the</span></span>
<span><span class="co"># current version of the GDC API. I have filed a feature</span></span>
<span><span class="co"># request to remove this requirement.</span></span>
<span><span class="va">resp</span> <span class="op">=</span> <span class="fu"><a href="../reference/query.html">cases</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/filtering.html">filter</a></span><span class="op">(</span><span class="op">~</span> <span class="va">project.project_id</span><span class="op">==</span><span class="st">'TCGA-BRCA'</span> <span class="op">&amp;</span></span>
<span>                              <span class="va">samples.sample_type</span><span class="op">==</span><span class="st">'Solid Tissue Normal'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="../reference/select.html">select</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="../reference/default_fields.html">default_fields</a></span><span class="op">(</span><span class="fu"><a href="../reference/query.html">cases</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>,<span class="st">'samples.sample_type'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/response.html">response_all</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/count.html">count</a></span><span class="op">(</span><span class="va">resp</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 162</span></span></code></pre>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">=</span> <span class="va">resp</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/results.html">results</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">res</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,list.len<span class="op">=</span><span class="fl">6</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of 1</span></span>
<span><span class="co">##  $ id: chr [1:162] "7875018b-91dc-4ab1-83b0-ee3883e2816e" "791c5768-f0f5-4ab6-86eb-998e5c4b49e3" "9ddc3e7b-8b54-4a83-8335-8053940f56c1" "3558dec7-eae3-4a41-a217-266c6a4535fb" ...</span></span></code></pre>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/ids.html">ids</a></span><span class="op">(</span><span class="va">resp</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "7875018b-91dc-4ab1-83b0-ee3883e2816e"</span></span>
<span><span class="co">## [2] "791c5768-f0f5-4ab6-86eb-998e5c4b49e3"</span></span>
<span><span class="co">## [3] "9ddc3e7b-8b54-4a83-8335-8053940f56c1"</span></span>
<span><span class="co">## [4] "3558dec7-eae3-4a41-a217-266c6a4535fb"</span></span>
<span><span class="co">## [5] "33fe5833-d7a0-43d7-a03d-56985ea448a6"</span></span>
<span><span class="co">## [6] "7317e605-93ba-49a9-8743-20613b02767a"</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="get-all-tcga-case-ids-that-are-female">Get all TCGA case ids that are female<a class="anchor" aria-label="anchor" href="#get-all-tcga-case-ids-that-are-female"></a>
</h4>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/query.html">cases</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="../reference/filtering.html">filter</a></span><span class="op">(</span><span class="op">~</span> <span class="va">project.program.name</span> <span class="op">==</span> <span class="st">'TCGA'</span> <span class="op">&amp;</span></span>
<span>    <span class="st">"cases.demographic.gender"</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="st">"female"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="../reference/results.html">results</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="../reference/ids.html">ids</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "3afa1e93-1df8-4e4c-aaa4-557463f4bb77"</span></span>
<span><span class="co">## [2] "b379bfb8-284e-4300-8325-85cfd6809cb8"</span></span>
<span><span class="co">## [3] "22eaba28-be67-4730-9846-860bc1fca29a"</span></span>
<span><span class="co">## [4] "1502c7d7-1535-4e56-9f34-30623acd50d5"</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="get-all-tcga-coad-case-ids-that-are-not-female">Get all TCGA-COAD case ids that are NOT female<a class="anchor" aria-label="anchor" href="#get-all-tcga-coad-case-ids-that-are-not-female"></a>
</h4>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/query.html">cases</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="../reference/filtering.html">filter</a></span><span class="op">(</span><span class="op">~</span> <span class="va">project.project_id</span> <span class="op">==</span> <span class="st">'TCGA-COAD'</span> <span class="op">&amp;</span></span>
<span>    <span class="st">"cases.demographic.gender"</span> <span class="op">%exclude%</span> <span class="st">"female"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="../reference/results.html">results</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="../reference/ids.html">ids</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "14efca0c-882d-4fba-b15c-64af6e213b80"</span></span>
<span><span class="co">## [2] "d7ba64d7-b4e5-488d-9f28-bcb7bf3c0f35"</span></span>
<span><span class="co">## [3] "1ad0b9b9-7ded-4281-b114-9dcc6138c037"</span></span>
<span><span class="co">## [4] "93edb105-5dc9-4a4a-9fca-c91f0161fcc7"</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="get-all-tcga-cases-that-are-missing-gender">Get all TCGA cases that are missing gender<a class="anchor" aria-label="anchor" href="#get-all-tcga-cases-that-are-missing-gender"></a>
</h4>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/query.html">cases</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="../reference/filtering.html">filter</a></span><span class="op">(</span><span class="op">~</span> <span class="va">project.program.name</span> <span class="op">==</span> <span class="st">'TCGA'</span> <span class="op">&amp;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/missing.html" class="external-link">missing</a></span><span class="op">(</span><span class="st">"cases.demographic.gender"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="../reference/results.html">results</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="../reference/ids.html">ids</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "bc66f21c-7ae6-4312-b6c1-ea3e19f90485"</span></span>
<span><span class="co">## [2] "a94de778-9c21-410d-8b9d-0f9240036bb8"</span></span>
<span><span class="co">## [3] "1e5053ae-9a78-41ed-81cd-1a43756fca09"</span></span>
<span><span class="co">## [4] "f8360744-6bc8-4f53-b1fc-a133789455a8"</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="get-all-tcga-cases-that-are-not-missing-gender">Get all TCGA cases that are NOT missing gender<a class="anchor" aria-label="anchor" href="#get-all-tcga-cases-that-are-not-missing-gender"></a>
</h4>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/query.html">cases</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="../reference/filtering.html">filter</a></span><span class="op">(</span><span class="op">~</span> <span class="va">project.program.name</span> <span class="op">==</span> <span class="st">'TCGA'</span> <span class="op">&amp;</span></span>
<span>    <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/missing.html" class="external-link">missing</a></span><span class="op">(</span><span class="st">"cases.demographic.gender"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="../reference/results.html">results</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="../reference/ids.html">ids</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "3afa1e93-1df8-4e4c-aaa4-557463f4bb77"</span></span>
<span><span class="co">## [2] "b379bfb8-284e-4300-8325-85cfd6809cb8"</span></span>
<span><span class="co">## [3] "22eaba28-be67-4730-9846-860bc1fca29a"</span></span>
<span><span class="co">## [4] "1502c7d7-1535-4e56-9f34-30623acd50d5"</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="files">Files<a class="anchor" aria-label="anchor" href="#files"></a>
</h3>
<div class="section level4">
<h4 id="how-many-of-each-type-of-file-are-available">How many of each type of file are available?<a class="anchor" aria-label="anchor" href="#how-many-of-each-type-of-file-are-available"></a>
</h4>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="../reference/query.html">files</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/faceting.html">facet</a></span><span class="op">(</span><span class="st">'type'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/aggregations.html">aggregations</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">res</span><span class="op">$</span><span class="va">type</span></span></code></pre></div>
<pre><code><span><span class="co">##    doc_count                           key</span></span>
<span><span class="co">## 1     222083    annotated_somatic_mutation</span></span>
<span><span class="co">## 2     114445                 aligned_reads</span></span>
<span><span class="co">## 3     102544       simple_somatic_mutation</span></span>
<span><span class="co">## 4      74909          structural_variation</span></span>
<span><span class="co">## 5      58881           copy_number_segment</span></span>
<span><span class="co">## 6      46151          copy_number_estimate</span></span>
<span><span class="co">## 7      37654               gene_expression</span></span>
<span><span class="co">## 8      33636   aggregated_somatic_mutation</span></span>
<span><span class="co">## 9      33234              mirna_expression</span></span>
<span><span class="co">## 10     31714      masked_methylation_array</span></span>
<span><span class="co">## 11     30075                   slide_image</span></span>
<span><span class="co">## 12     26001        biospecimen_supplement</span></span>
<span><span class="co">## 13     15857        methylation_beta_value</span></span>
<span><span class="co">## 14     15633       masked_somatic_mutation</span></span>
<span><span class="co">## 15     13133           clinical_supplement</span></span>
<span><span class="co">## 16      7906            protein_expression</span></span>
<span><span class="co">## 17        67              pathology_report</span></span>
<span><span class="co">## 18        54 secondary_expression_analysis</span></span></code></pre>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">type</span>,<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">key</span>,y <span class="op">=</span> <span class="va">doc_count</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat<span class="op">=</span><span class="st">'identity'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="overview_files/figure-html/filesVCFCount-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="find-gene-level-rna-seq-quantification-files-for-gbm">Find gene-level RNA-seq quantification files for GBM<a class="anchor" aria-label="anchor" href="#find-gene-level-rna-seq-quantification-files-for-gbm"></a>
</h4>
<div class="sourceCode" id="cb93"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">q</span> <span class="op">=</span> <span class="fu"><a href="../reference/query.html">files</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="../reference/select.html">select</a></span><span class="op">(</span><span class="fu"><a href="../reference/available_fields.html">available_fields</a></span><span class="op">(</span><span class="st">'files'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/filtering.html">filter</a></span><span class="op">(</span><span class="op">~</span> <span class="va">cases.project.project_id</span><span class="op">==</span><span class="st">'TCGA-GBM'</span> <span class="op">&amp;</span></span>
<span>               <span class="va">data_type</span><span class="op">==</span><span class="st">'Gene Expression Quantification'</span><span class="op">)</span></span>
<span><span class="va">q</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/faceting.html">facet</a></span><span class="op">(</span><span class="st">'analysis.workflow_type'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/aggregations.html">aggregations</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## list()</span></span></code></pre>
<div class="sourceCode" id="cb95"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># so need to add another filter</span></span>
<span><span class="va">file_ids</span> <span class="op">=</span> <span class="va">q</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/filtering.html">filter</a></span><span class="op">(</span><span class="op">~</span> <span class="va">cases.project.project_id</span><span class="op">==</span><span class="st">'TCGA-GBM'</span> <span class="op">&amp;</span></span>
<span>                            <span class="va">data_type</span><span class="op">==</span><span class="st">'Gene Expression Quantification'</span> <span class="op">&amp;</span></span>
<span>                            <span class="va">analysis.workflow_type</span> <span class="op">==</span> <span class="st">'STAR - Counts'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="../reference/select.html">select</a></span><span class="op">(</span><span class="st">'file_id'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/response.html">response_all</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/ids.html">ids</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="slicing">Slicing<a class="anchor" aria-label="anchor" href="#slicing"></a>
</h3>
<div class="section level4">
<h4 id="get-all-bam-file-ids-from-tcga-gbm">Get all BAM file ids from TCGA-GBM<a class="anchor" aria-label="anchor" href="#get-all-bam-file-ids-from-tcga-gbm"></a>
</h4>
<p><strong>I need to figure out how to do slicing reproducibly in a testing environment and for vignette building</strong>.</p>
<div class="sourceCode" id="cb96"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">q</span> <span class="op">=</span> <span class="fu"><a href="../reference/query.html">files</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="../reference/select.html">select</a></span><span class="op">(</span><span class="fu"><a href="../reference/available_fields.html">available_fields</a></span><span class="op">(</span><span class="st">'files'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/filtering.html">filter</a></span><span class="op">(</span><span class="op">~</span> <span class="va">cases.project.project_id</span> <span class="op">==</span> <span class="st">'TCGA-GBM'</span> <span class="op">&amp;</span></span>
<span>               <span class="va">data_type</span> <span class="op">==</span> <span class="st">'Aligned Reads'</span> <span class="op">&amp;</span></span>
<span>               <span class="va">experimental_strategy</span> <span class="op">==</span> <span class="st">'RNA-Seq'</span> <span class="op">&amp;</span></span>
<span>               <span class="va">data_format</span> <span class="op">==</span> <span class="st">'BAM'</span><span class="op">)</span></span>
<span><span class="va">file_ids</span> <span class="op">=</span> <span class="va">q</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/response.html">response_all</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/ids.html">ids</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb97"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bamfile</span> <span class="op">=</span> <span class="fu"><a href="../reference/slicing.html">slicing</a></span><span class="op">(</span><span class="va">file_ids</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,regions<span class="op">=</span><span class="st">"chr12:6534405-6538375"</span>,token<span class="op">=</span><span class="fu"><a href="../reference/gdc_token.html">gdc_token</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicAlignments" class="external-link">GenomicAlignments</a></span><span class="op">)</span></span>
<span><span class="va">aligns</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicAlignments/man/readGAlignments.html" class="external-link">readGAlignments</a></span><span class="op">(</span><span class="va">bamfile</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="troubleshooting">Troubleshooting<a class="anchor" aria-label="anchor" href="#troubleshooting"></a>
</h2>
<div class="section level3">
<h3 id="ssl-connection-errors">SSL connection errors<a class="anchor" aria-label="anchor" href="#ssl-connection-errors"></a>
</h3>
<ul>
<li>Symptom: Trying to connect to the API results in:</li>
</ul>
<pre><code>Error in curl::curl_fetch_memory(url, handle = handle) :
SSL connect error</code></pre>
<ul>
<li>Possible solutions: The <a href="http://stackoverflow.com/a/42599546/459633" class="external-link">issue is that the GDC supports only recent security Transport Layer Security (TLS)</a>, so the only known fix is to upgrade the system <code>openssl</code> to version 1.0.1 or later.
<ul>
<li>
<a href="https://github.com/Bioconductor/GenomicDataCommons/issues/35#issuecomment-284233510" class="external-link">[Mac OS]</a>,</li>
<li><a href="http://askubuntu.com/a/434245" class="external-link">[Ubuntu]</a></li>
<li>
<a href="https://www.liquidweb.com/kb/update-and-patch-openssl-for-the-ccs-injection-vulnerability/" class="external-link">[Centos/RHEL]</a>. After upgrading <code>openssl</code>, reinstall the R <code>curl</code> and <code>httr</code> packages.</li>
</ul>
</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="sessioninfo">sessionInfo()<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<div class="sourceCode" id="cb99"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 20.04.4 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0</span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] ggplot2_3.3.6             GenomicDataCommons_1.21.2</span></span>
<span><span class="co">## [3] magrittr_2.0.3            knitr_1.39               </span></span>
<span><span class="co">## [5] BiocStyle_2.24.0         </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] assertthat_0.2.1       rprojroot_2.0.3        digest_0.6.29         </span></span>
<span><span class="co">##  [4] utf8_1.2.2             R6_2.5.1               GenomeInfoDb_1.32.2   </span></span>
<span><span class="co">##  [7] stats4_4.2.1           evaluate_0.15          highr_0.9             </span></span>
<span><span class="co">## [10] httr_1.4.3             pillar_1.8.0           zlibbioc_1.42.0       </span></span>
<span><span class="co">## [13] rlang_1.0.4            curl_4.3.2             jquerylib_0.1.4       </span></span>
<span><span class="co">## [16] S4Vectors_0.34.0       rmarkdown_2.14         pkgdown_2.0.6         </span></span>
<span><span class="co">## [19] labeling_0.4.2         textshaping_0.3.6      desc_1.4.1            </span></span>
<span><span class="co">## [22] readr_2.1.2            stringr_1.4.0          munsell_0.5.0         </span></span>
<span><span class="co">## [25] RCurl_1.98-1.8         bit_4.0.4              compiler_4.2.1        </span></span>
<span><span class="co">## [28] xfun_0.31              pkgconfig_2.0.3        systemfonts_1.0.4     </span></span>
<span><span class="co">## [31] BiocGenerics_0.42.0    htmltools_0.5.3        tidyselect_1.1.2      </span></span>
<span><span class="co">## [34] tibble_3.1.8           GenomeInfoDbData_1.2.8 bookdown_0.27         </span></span>
<span><span class="co">## [37] IRanges_2.30.0         fansi_1.0.3            crayon_1.5.1          </span></span>
<span><span class="co">## [40] dplyr_1.0.9            tzdb_0.3.0             withr_2.5.0           </span></span>
<span><span class="co">## [43] bitops_1.0-7           rappdirs_0.3.3         grid_4.2.1            </span></span>
<span><span class="co">## [46] gtable_0.3.0           jsonlite_1.8.0         lifecycle_1.0.1       </span></span>
<span><span class="co">## [49] DBI_1.1.3              scales_1.2.0           cli_3.3.0             </span></span>
<span><span class="co">## [52] stringi_1.7.8          vroom_1.5.7            cachem_1.0.6          </span></span>
<span><span class="co">## [55] farver_2.1.1           XVector_0.36.0         fs_1.5.2              </span></span>
<span><span class="co">## [58] xml2_1.3.3             bslib_0.4.0            ellipsis_0.3.2        </span></span>
<span><span class="co">## [61] ragg_1.2.2             generics_0.1.3         vctrs_0.4.1           </span></span>
<span><span class="co">## [64] tools_4.2.1            bit64_4.0.5            glue_1.6.2            </span></span>
<span><span class="co">## [67] purrr_0.3.4            hms_1.1.1              parallel_4.2.1        </span></span>
<span><span class="co">## [70] fastmap_1.1.0          yaml_2.3.5             colorspace_2.0-3      </span></span>
<span><span class="co">## [73] BiocManager_1.30.18    GenomicRanges_1.48.0   memoise_2.0.1         </span></span>
<span><span class="co">## [76] sass_0.4.2</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="developer-notes">Developer notes<a class="anchor" aria-label="anchor" href="#developer-notes"></a>
</h2>
<ul>
<li>The <code>S3</code> object-oriented programming paradigm is used.</li>
<li>We have adopted a functional programming style with functions and methods that often take an “object” as the first argument. This style lends itself to pipeline-style programming.</li>
<li>The GenomicDataCommons package uses the <a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#alternative-request-format" class="external-link">alternative request format (POST)</a> to allow very large request bodies.</li>
</ul>
</div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Martin Morgan, Sean Davis.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
