<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="GenomicDataCommons">
<title>Questions and answers from over the years • GenomicDataCommons</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Questions and answers from over the years">
<meta property="og:description" content="GenomicDataCommons">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">GenomicDataCommons</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.21.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/overview.html">The GenomicDataCommons Package</a>
    <a class="dropdown-item" href="../articles/questions-and-answers.html">Questions and answers from over the years</a>
    <a class="dropdown-item" href="../articles/somatic_mutations.html">Working with simple somatic mutations</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/Bioconductor/GenomicDataCommons/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="questions-and-answers_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Questions and answers from over the years</h1>
                        <h4 data-toc-skip class="author">Sean Davis</h4>
            
            <h4 data-toc-skip class="date">Thursday, April 28, 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/Bioconductor/GenomicDataCommons/blob/HEAD/vignettes/questions-and-answers.Rmd" class="external-link"><code>vignettes/questions-and-answers.Rmd</code></a></small>
      <div class="d-none name"><code>questions-and-answers.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="how-could-i-generate-a-manifest-file-with-filtering-of-race-and-ethnicity">How could I generate a manifest file with filtering of Race and Ethnicity?<a class="anchor" aria-label="anchor" href="#how-could-i-generate-a-manifest-file-with-filtering-of-race-and-ethnicity"></a>
</h2>
<p>From <a href="https://support.bioconductor.org/p/9138939/" class="external-link uri">https://support.bioconductor.org/p/9138939/</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicDataCommons" class="external-link">GenomicDataCommons</a></span>,quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">## Attaching package: 'GenomicDataCommons'</span></code></pre>
<pre><code><span class="co">## The following object is masked from 'package:stats':</span>
<span class="co">## </span>
<span class="co">##     filter</span></code></pre>
<p>I made a small change to the filtering expression approach based on changes to lazy evaluation best practices. There is now no need to include the <code>~</code> in the filter expression. So:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">q</span> <span class="op">=</span> <span class="fu"><a href="../reference/query.html">files</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="../reference/filtering.html">filter</a></span><span class="op">(</span>
    <span class="va">cases.project.project_id</span> <span class="op">==</span> <span class="st">'TCGA-COAD'</span> <span class="op">&amp;</span>
      <span class="va">data_type</span> <span class="op">==</span> <span class="st">'Aligned Reads'</span> <span class="op">&amp;</span>
      <span class="va">experimental_strategy</span> <span class="op">==</span> <span class="st">'RNA-Seq'</span> <span class="op">&amp;</span>
      <span class="va">data_format</span> <span class="op">==</span> <span class="st">'BAM'</span><span class="op">)</span></code></pre></div>
<p>And get a count of the results:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/count.html">count</a></span><span class="op">(</span><span class="va">q</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 1183</span></code></pre>
<p>And the manifest.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/manifest.html">manifest</a></span><span class="op">(</span><span class="va">q</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## <span style="color: #949494;"># A tibble: 1,183 × 5</span></span>
<span class="co">##    id                                   filename             md5      size state</span>
<span class="co">##  <span style="color: #BCBCBC;">*</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 1</span> 0638c43d-ace4-4acf-8a99-b1cbe34b73d4 e074ff82-1def-4bfc-… 3899… 9.19<span style="color: #949494;">e</span> 7 rele…</span>
<span class="co">## <span style="color: #BCBCBC;"> 2</span> e730e97c-c9af-45a1-9bed-3ea3d5a31075 1a7a23fc-67d0-47d7-… f40e… 8.75<span style="color: #949494;">e</span> 9 rele…</span>
<span class="co">## <span style="color: #BCBCBC;"> 3</span> 707f452a-a23e-4d2b-96d2-5e9101f6ff5c e074ff82-1def-4bfc-… 51c0… 6.42<span style="color: #949494;">e</span> 9 rele…</span>
<span class="co">## <span style="color: #BCBCBC;"> 4</span> fea4d7c0-1eeb-466b-9b58-8e91535f3e3d 90242448-2567-4d5b-… 0872… 7.11<span style="color: #949494;">e</span> 7 rele…</span>
<span class="co">## <span style="color: #BCBCBC;"> 5</span> fa10a537-7293-4803-844f-e0f9273eea15 2db6d1af-ff0c-468f-… 1ab9… 9.77<span style="color: #949494;">e</span> 9 rele…</span>
<span class="co">## <span style="color: #BCBCBC;"> 6</span> 95d17a59-f986-41a6-906b-f8e7d7c93649 bf93be3e-d4f1-4efb-… 0b79… 2.28<span style="color: #949494;">e</span> 9 rele…</span>
<span class="co">## <span style="color: #BCBCBC;"> 7</span> 64dc1cf1-30cd-4780-87d4-a0a53d2b8e71 88a33f23-72a0-439b-… 39b0… 1.20<span style="color: #949494;">e</span>10 rele…</span>
<span class="co">## <span style="color: #BCBCBC;"> 8</span> 79ec6ebe-5854-4431-86cc-346ac8632626 88a33f23-72a0-439b-… 0f63… 6.83<span style="color: #949494;">e</span> 7 rele…</span>
<span class="co">## <span style="color: #BCBCBC;"> 9</span> a2b0827d-05d6-4611-8e01-359547b11f57 8fba9293-e3f1-41ef-… 61bd… 1.51<span style="color: #949494;">e</span>10 rele…</span>
<span class="co">## <span style="color: #BCBCBC;">10</span> aea7eb73-bbb3-423a-8477-69464d8f46f3 8fba9293-e3f1-41ef-… 33be… 8.08<span style="color: #949494;">e</span> 9 rele…</span>
<span class="co">## <span style="color: #949494;"># … with 1,173 more rows</span></span></code></pre>
<p>Your question about race and ethnicity is a good one.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">all_fields</span> <span class="op">=</span> <span class="fu"><a href="../reference/available_fields.html">available_fields</a></span><span class="op">(</span><span class="fu"><a href="../reference/query.html">files</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>And we can grep for <code>race</code> or <code>ethnic</code> to get potential matching fields to look at.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">'race|ethnic'</span>,<span class="va">all_fields</span>,value<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "cases.demographic.ethnicity"                 </span>
<span class="co">## [2] "cases.demographic.race"                      </span>
<span class="co">## [3] "cases.follow_ups.hormonal_contraceptive_type"</span>
<span class="co">## [4] "cases.follow_ups.hormonal_contraceptive_use" </span>
<span class="co">## [5] "cases.follow_ups.scan_tracer_used"</span></code></pre>
<p>Now, we can check available values for each field to determine how to complete our filter expressions.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/available_values.html">available_values</a></span><span class="op">(</span><span class="st">'files'</span>,<span class="st">"cases.demographic.ethnicity"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "not hispanic or latino" "not reported"           "hispanic or latino"    </span>
<span class="co">## [4] "unknown"                "not allowed to collect" "_missing"</span></code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/available_values.html">available_values</a></span><span class="op">(</span><span class="st">'files'</span>,<span class="st">"cases.demographic.race"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##  [1] "white"                                    </span>
<span class="co">##  [2] "not reported"                             </span>
<span class="co">##  [3] "black or african american"                </span>
<span class="co">##  [4] "asian"                                    </span>
<span class="co">##  [5] "unknown"                                  </span>
<span class="co">##  [6] "other"                                    </span>
<span class="co">##  [7] "not allowed to collect"                   </span>
<span class="co">##  [8] "american indian or alaska native"         </span>
<span class="co">##  [9] "native hawaiian or other pacific islander"</span>
<span class="co">## [10] "_missing"</span></code></pre>
<p>We can complete our filter expression now to limit to <code>white</code> race only.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">q_white_only</span> <span class="op">=</span> <span class="va">q</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="../reference/filtering.html">filter</a></span><span class="op">(</span><span class="va">cases.demographic.race</span><span class="op">==</span><span class="st">'white'</span><span class="op">)</span>
<span class="fu"><a href="../reference/count.html">count</a></span><span class="op">(</span><span class="va">q_white_only</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 691</span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/manifest.html">manifest</a></span><span class="op">(</span><span class="va">q_white_only</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## <span style="color: #949494;"># A tibble: 691 × 5</span></span>
<span class="co">##    id                                   filename             md5      size state</span>
<span class="co">##  <span style="color: #BCBCBC;">*</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 1</span> e730e97c-c9af-45a1-9bed-3ea3d5a31075 1a7a23fc-67d0-47d7-… f40e… 8.75<span style="color: #949494;">e</span> 9 rele…</span>
<span class="co">## <span style="color: #BCBCBC;"> 2</span> fea4d7c0-1eeb-466b-9b58-8e91535f3e3d 90242448-2567-4d5b-… 0872… 7.11<span style="color: #949494;">e</span> 7 rele…</span>
<span class="co">## <span style="color: #BCBCBC;"> 3</span> a2b0827d-05d6-4611-8e01-359547b11f57 8fba9293-e3f1-41ef-… 61bd… 1.51<span style="color: #949494;">e</span>10 rele…</span>
<span class="co">## <span style="color: #BCBCBC;"> 4</span> aea7eb73-bbb3-423a-8477-69464d8f46f3 8fba9293-e3f1-41ef-… 33be… 8.08<span style="color: #949494;">e</span> 9 rele…</span>
<span class="co">## <span style="color: #BCBCBC;"> 5</span> 980b340a-71e2-4e31-b41b-dc273ffcd635 8fba9293-e3f1-41ef-… 97b5… 1.01<span style="color: #949494;">e</span> 8 rele…</span>
<span class="co">## <span style="color: #BCBCBC;"> 6</span> 1f41925d-d70a-470c-998f-35d6b28ae099 63bab58e-1dfd-43df-… 3c4e… 2.10<span style="color: #949494;">e</span>10 rele…</span>
<span class="co">## <span style="color: #BCBCBC;"> 7</span> 6f4370b2-6c7f-4550-83ec-cc9ab50a34b7 63bab58e-1dfd-43df-… 10d6… 9.55<span style="color: #949494;">e</span> 7 rele…</span>
<span class="co">## <span style="color: #BCBCBC;"> 8</span> 4dbe8520-3ca8-4d0e-ad9a-0fd794d0e261 63bab58e-1dfd-43df-… 3ace… 9.70<span style="color: #949494;">e</span> 9 rele…</span>
<span class="co">## <span style="color: #BCBCBC;"> 9</span> 85535fab-ba49-4b3e-b372-08c15a997042 f40a9044-a36a-4091-… 6b0d… 1.23<span style="color: #949494;">e</span>10 rele…</span>
<span class="co">## <span style="color: #BCBCBC;">10</span> 008b411e-7de8-46f2-9ad8-185cd49ee2e6 f40a9044-a36a-4091-… 9598… 5.84<span style="color: #949494;">e</span> 9 rele…</span>
<span class="co">## <span style="color: #949494;"># … with 681 more rows</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="how-can-i-get-the-number-of-cases-with-rna-seq-data-added-by-date-to-tcga-project-with-genomicdatacommons">How can I get the number of cases with RNA-Seq data added by date to TCGA project with <code>GenomicDataCommons</code>?<a class="anchor" aria-label="anchor" href="#how-can-i-get-the-number-of-cases-with-rna-seq-data-added-by-date-to-tcga-project-with-genomicdatacommons"></a>
</h2>
<ul>
<li>From <a href="https://support.bioconductor.org/p/9135791/" class="external-link uri">https://support.bioconductor.org/p/9135791/</a>
</li>
</ul>
<p>I would like to get the number of cases added (created, any logical datetime would suffice here) to the TCGA project by experiment type. I attempted to get this data via GenomicDataCommons package, but it is giving me I believe the number of files for a given experiment type rather than number cases. How can I get the number of cases for which there is RNA-Seq data?</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">## Attaching package: 'dplyr'</span></code></pre>
<pre><code><span class="co">## The following objects are masked from 'package:GenomicDataCommons':</span>
<span class="co">## </span>
<span class="co">##     count, filter, select</span></code></pre>
<pre><code><span class="co">## The following objects are masked from 'package:stats':</span>
<span class="co">## </span>
<span class="co">##     filter, lag</span></code></pre>
<pre><code><span class="co">## The following objects are masked from 'package:base':</span>
<span class="co">## </span>
<span class="co">##     intersect, setdiff, setequal, union</span></code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicDataCommons" class="external-link">GenomicDataCommons</a></span><span class="op">)</span>

<span class="fu"><a href="../reference/query.html">cases</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu">GenomicDataCommons</span><span class="fu">::</span><span class="fu"><a href="../reference/filtering.html">filter</a></span><span class="op">(</span><span class="op">~</span> <span class="va">project.program.name</span><span class="op">==</span><span class="st">'TCGA'</span> <span class="op">&amp;</span> 
                               <span class="va">files.experimental_strategy</span><span class="op">==</span><span class="st">'RNA-Seq'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="../reference/faceting.html">facet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"files.created_datetime"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="../reference/aggregations.html">aggregations</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
  <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">key</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## <span style="color: #949494;"># A tibble: 200 × 2</span></span>
<span class="co">##    doc_count key                             </span>
<span class="co">##        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                           </span>
<span class="co">## <span style="color: #BCBCBC;"> 1</span>       362 2021-04-05t12:48:23.926301-05:00</span>
<span class="co">## <span style="color: #BCBCBC;"> 2</span>       438 2021-04-05t08:30:00.775501-05:00</span>
<span class="co">## <span style="color: #BCBCBC;"> 3</span>       374 2021-04-05t08:29:15.674486-05:00</span>
<span class="co">## <span style="color: #BCBCBC;"> 4</span>       305 2021-04-05t08:26:08.920845-05:00</span>
<span class="co">## <span style="color: #BCBCBC;"> 5</span>       349 2021-04-05t08:22:48.913195-05:00</span>
<span class="co">## <span style="color: #BCBCBC;"> 6</span>       351 2021-04-05t08:21:59.240799-05:00</span>
<span class="co">## <span style="color: #BCBCBC;"> 7</span>       269 2021-04-05t08:21:27.725962-05:00</span>
<span class="co">## <span style="color: #BCBCBC;"> 8</span>       325 2021-04-05t08:20:56.849817-05:00</span>
<span class="co">## <span style="color: #BCBCBC;"> 9</span>       428 2021-04-05t08:20:25.746896-05:00</span>
<span class="co">## <span style="color: #BCBCBC;">10</span>       214 2021-04-05t08:19:48.312949-05:00</span>
<span class="co">## <span style="color: #949494;"># … with 190 more rows</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Martin Morgan, Sean Davis.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
